<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CAESAR: a cross-technology and cross-resolution framework for spatial omics annotation • CAESAR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="CAESAR: a cross-technology and cross-resolution framework for spatial omics annotation">
<meta property="og:description" content="Biotechnology in spatial omics has advanced rapidly over the past few years, enhancing both throughput and resolution. However, existing annotation pipelines in spatial omics predominantly rely on clustering methods, lacking the flexibility to integrate extensive annotated information from single-cell RNA sequencing (scRNA-seq) due to discrepancies in spatial resolutions, species, or modalities. Here we introduce the CAESAR suite, an open-source software package that provides image-based spatial co-embedding of locations and genomic features. It uniquely transfers labels from scRNA-seq reference, enabling the annotation of spatial omics datasets across different technologies, resolutions, species, and modalities, based on the conserved relationship between signature genes and cells/locations at an appropriate level of granularity. Notably, CAESAR enriches location-level pathways, allowing for the detection of gradual biological pathway activation within spatially defined domain types.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CAESAR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/STMOB.html">ST MOB data analysis</a>
    </li>
    <li>
      <a href="articles/XeniumBC.html">Xenium BC data analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/XiaoZhangryy/CAESAR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="caesar">CAESAR<a class="anchor" aria-label="anchor" href="#caesar"></a>
</h1></div>
<p>========================================================================= <!-- badges: start --></p>
<p><a href="https://cran.r-project.org/package=CAESAR" class="external-link"><img src="https://www.r-pkg.org/badges/version-ago/CAESAR"></a> <a href="https://cran.r-project.org/package=CAESAR" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/CAESAR?color=orange"></a> <a href="https://cran.r-project.org/package=CAESAR" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/grand-total/CAESAR?color=orange"></a> <!-- badges: end --></p>
<p>CAESAR suite is an open-source software package that provides image-based spatial co-embedding of locations and genomic features. The ‘CAESAR’ package is specifically developed by the Jin Liu’s lab for annotation and enrichment analysis of spatially resolved transcriptomics (SRT) dataset. It uniquely transfers labels from scRNA-seq reference, enabling the annotation of spatial omics datasets across different technologies, resolutions, species, and modalities, based on the conserved relationship between signature genes and cells/locations at an appropriate level of granularity. Notably, CAESAR enriches location-level pathways, allowing for the detection of gradual biological pathway activation within spatially defined domain types.</p>
<p>Check out our <a href="https://XiaoZhangryy.github.io/CAESAR/index.html" class="external-link">Package Website</a> for a more complete description of the methods and analyses.</p>
<p>CAESAR provides image-based spatial aware co-embedding of locations and genomic features.</p>
<p>By assuming a conserved relationship between genomic features and cells/locations within each cell/domain type at an appropriate level of granularity, the CAESAR suite flexibly annotates spatial omics datasets in a variety of contexts, for instance:</p>
<ul>
<li>Using multiple reference datasets.</li>
<li>Across species.</li>
<li>Across resolutions.</li>
<li>Across modalities.</li>
<li>Across technologies.</li>
</ul>
<p>The CAESAR suite includes functions for hypothesis testing to identify pathways enriched in each cell/location or cell/domain type. For instance:</p>
<ul>
<li>Test whether pathways are enriched in dataset.</li>
<li>Calculate pathway enrichment score for each cell/location.</li>
<li>Detect cell/domain type differentially enriched pathway.</li>
</ul>
<p>In addition, once the co-embeddings of (multiple) dataset are estimated by CAESAR, the package provides functionality for further data exploration, analysis, and visualization. Users can:</p>
<ul>
<li>Detect the signature genes .</li>
<li>Determine appropriate marker gene sets based on signature gene lists obtained from reference datasets</li>
<li>Recover comparable gene expression matrices among datasets .</li>
<li>Integrate signature gene lists from multiple datasets.</li>
<li>Visualize the co-embeddings on UMAP space.</li>
<li>Visualize the signature genes on UMAP space.</li>
</ul>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p>“CAESAR” depends on the ‘Rcpp’ and ‘RcppArmadillo’ package, which requires appropriate setup of computer. For the users that have set up system properly for compiling C++ files, the following installation command will work. ```{Rmd} # Method 1: install CAESAR from CRAN install.packages(‘CAESAR’)</p>
</div>
<div class="section level1">
<h1 id="for-the-newest-version-of-caesar-users-can-use-method-2-for-installation">For the newest version of CAESAR, users can use method 2 for installation.<a class="anchor" aria-label="anchor" href="#for-the-newest-version-of-caesar-users-can-use-method-2-for-installation"></a>
</h1>
</div>
<div class="section level1">
<h1 id="method-2-install-caesar-from-github">Method 2: Install CAESAR from Github<a class="anchor" aria-label="anchor" href="#method-2-install-caesar-from-github"></a>
</h1>
<p>if (!require(“remotes”, quietly = TRUE)) install.packages(“remotes”) remotes::install_github(“XiaoZhangryy/CAESAR”)</p>
</div>
<div class="section level1">
<h1 id="if-some-dependent-packages-such-as-scater-on-bioconductor-can-not-be-installed-nomrally-use-following-commands-then-run-abouve-command">If some dependent packages (such as <code>scater</code>) on Bioconductor can not be installed nomrally, use following commands, then run abouve command.<a class="anchor" aria-label="anchor" href="#if-some-dependent-packages-such-as-scater-on-bioconductor-can-not-be-installed-nomrally-use-following-commands-then-run-abouve-command"></a>
</h1>
<p>if (!require(“BiocManager”, quietly = TRUE)) ## install BiocManager install.packages(“BiocManager”) # install the package on Bioconducter BiocManager::install(c(“scater”))</p>
<pre><code>
## Usage
For usage examples and guided walkthroughs, check the `vignettes` directory of the repo.

Tutorials for CAESAR suite:

* [Analysis for Xenium breast cancer data](https://XiaoZhangryy.github.io/CAESAR/articles/XeniumBC.html)
* [Analysis for ST mouse olfactory bulb data](https://XiaoZhangryy.github.io/CAESAR/articles/STMOB.html)

For the users that don't have set up system properly, the following setup on different systems can be referred.
## Setup on Windows system
First, download [Rtools](https://cran.r-project.org/bin/windows/Rtools/); second, add the Rtools directory to the environment variable.

## Setup on MacOS system
First, install Xcode. Installation about Xcode can be referred [here](https://stackoverflow.com/questions/8291146/xcode-installation-on-mac).

Second, install "gfortran" for compiling C++ and Fortran at [here](https://github.com/fxcoudert/gfortran-for-macOS).

## Setup on Linux  system
If you use conda environment on Linux system and some dependent packages (such as `scater`) can not normally installed, you can search R package at anaconda.org website. We take the `scater` package as example, and its search result is https://anaconda.org/bioconda/bioconductor-scater. Then you can install it in conda environment by following command.
```{Linux}

conda install -c bioconda bioconductor-scater</code></pre>
<p>For the user not using conda environment, if dependent packages (such as <code>scater</code>) not normally installed are in Bioconductor, then use the following command to install the dependent packages. <code>{Linux} # install BiocManager if (!require("BiocManager", quietly = TRUE))     install.packages("BiocManager") # install the package on Bioconducter BiocManager::install(c("scater"))</code> If dependent packages (such as <code>ProFAST</code>) not normally installed are in CRAN, then use the following command to install the dependent packages. <code>{Linux} # install the package on CRAN install.packages("ProFAST")</code></p>
</div>
<div class="section level1">
<h1 id="demonstration">Demonstration<a class="anchor" aria-label="anchor" href="#demonstration"></a>
</h1>
<p>For an example of typical CAESAR usage, please see our <a href="https://XiaoZhangryy.github.io/CAESAR/index.html" class="external-link">Package Website</a> for a demonstration and overview of the functions included in CAESAR.</p>
</div>
<div class="section level1">
<h1 id="news">NEWs<a class="anchor" aria-label="anchor" href="#news"></a>
</h1>
<ul>
<li>CAESAR version 0.1 (2024-08-30)</li>
</ul>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/XiaoZhangryy/CAESAR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/XiaoZhangryy/CAESAR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>GPL (&gt;= 2)</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CAESAR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Xiao Zhang <br><small class="roles"> Author, maintainer </small>  </li>
<li>Wei Liu <br><small class="roles"> Author </small>  </li>
<li>Jin Liu <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Xiao Zhang, Wei Liu, Jin Liu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
