<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ST MOB data analysis • CAESAR.Suite</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="ST MOB data analysis">
<meta property="og:description" content="CAESAR.Suite">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CAESAR.Suite</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/STMOB.html">ST MOB data analysis</a>
    </li>
    <li>
      <a href="../articles/XeniumBC.html">Xenium BC data analysis</a>
    </li>
    <li>
      <a href="../articles/XeniumBCEIF.html">Xenium BC extract image features</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/XiaoZhangryy/CAESAR.Suite/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>ST MOB data analysis</h1>
                        <h4 data-toc-skip class="author">Xiao Zhang</h4>
            
            <h4 data-toc-skip class="date">2024-09-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/XiaoZhangryy/CAESAR.Suite/blob/HEAD/vignettes/STMOB.Rmd" class="external-link"><code>vignettes/STMOB.Rmd</code></a></small>
      <div class="hidden name"><code>STMOB.Rmd</code></div>

    </div>

    
    
<p>This vignette introduces the CAESAR.Suite workflow for the analysis
of MOB ST spatial transcriptomics dataset. In this vignette, the
workflow of CAESAR.Suite consists of four steps</p>
<ul>
<li>Reference dataset and target dataset preprocessing</li>
<li>Detect signature genes as cell type markers from scRNA-seq reference
data</li>
<li>Annotate MOB ST data using CAESAR</li>
<li>Downstream analysis (i.e. , signature gene analysis, visualization
of cell types and coembeddings)</li>
</ul>
<div class="section level2">
<h2 id="load-and-quality-control-both-reference-and-target-data">Load and quality control both reference and target data<a class="anchor" aria-label="anchor" href="#load-and-quality-control-both-reference-and-target-data"></a>
</h2>
<p>We demonstrate the use of CAESAR to MOB data, which can be downloaded
and load to the current working path by the following command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/XiaoZhangryy/CAESAR.Suite/blob/master/vignettes_data/MOB_ST.rda?raw=true"</span></span>
<span><span class="va">MOB_ST_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"MOB_ST.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="va">MOB_ST_file</span>, mode<span class="op">=</span><span class="st">'wb'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">MOB_ST_file</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 16034 features across 282 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (16034 features, 0 variable features)</span></span>
<span></span>
<span><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/XiaoZhangryy/CAESAR.Suite/blob/master/vignettes_data/MOB_scRNA.rda?raw=true"</span></span>
<span><span class="va">MOB_scRNA_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"MOB_scRNA.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="va">MOB_scRNA_file</span>, mode<span class="op">=</span><span class="st">'wb'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">MOB_scRNA_file</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">)</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 18560 features across 12801 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (18560 features, 0 variable features)</span></span></code></pre></div>
<p>The package can be loaded with the command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="co"># set a random seed for reproducibility.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/XiaoZhangryy/CAESAR.Suite" class="external-link">CAESAR.Suite</a></span><span class="op">)</span> <span class="co"># load the package of CAESAR method</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/feiyoung/ProFAST" class="external-link">ProFAST</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'ProFAST' was built under R version 4.3.3</span></span>
<span><span class="co">#&gt; Loading required package: gtools</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<p>Users can perform appropriate quality control on the reference
dataset and target data. Genes expressed in less than one cell are
required to remove to avoid unknown errors. Other quality control steps
can be set by the user according to the data quality. Here, cells with
less than five genes and genes expressed in less than one cell were
excluded.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MOB_ST</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span></span>
<span>    counts <span class="op">=</span> <span class="va">MOB_ST</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span>,</span>
<span>    meta.data <span class="op">=</span> <span class="va">MOB_ST</span><span class="op">@</span><span class="va">meta.data</span>,</span>
<span>    min.features <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    min.cells <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 16034 features across 280 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (16034 features, 0 variable features)</span></span>
<span></span>
<span><span class="va">MOB_scRNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span></span>
<span>    counts <span class="op">=</span> <span class="va">MOB_scRNA</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span>,</span>
<span>    meta.data <span class="op">=</span> <span class="va">MOB_scRNA</span><span class="op">@</span><span class="va">meta.data</span>,</span>
<span>    min.features <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    min.cells <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">)</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 17812 features across 12801 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (17812 features, 0 variable features)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocessing-and-align-reference-and-target-data">Preprocessing and align reference and target data<a class="anchor" aria-label="anchor" href="#preprocessing-and-align-reference-and-target-data"></a>
</h2>
<p>First, we normalize the data and select the variable genes. We align
genes and variable genes of reference and target data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># align genes</span></span>
<span><span class="va">common_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">MOB_ST</span> <span class="op">&lt;-</span> <span class="va">MOB_ST</span><span class="op">[</span><span class="va">common_genes</span>, <span class="op">]</span></span>
<span><span class="va">MOB_scRNA</span> <span class="op">&lt;-</span> <span class="va">MOB_scRNA</span><span class="op">[</span><span class="va">common_genes</span>, <span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">common_genes</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 13895</span></span>
<span></span>
<span><span class="va">MOB_ST</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span></span>
<span><span class="va">MOB_ST</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">MOB_ST</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span>
<span></span>
<span><span class="va">MOB_scRNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">)</span></span>
<span><span class="va">MOB_scRNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">MOB_scRNA</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span>
<span></span>
<span><span class="va">common_vgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">common_vgs</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">common_vgs</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">common_vgs</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 508</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="detect-signature-genes-for-each-cell-type-using-scrna-seq-reference-data">Detect signature genes for each cell type using scRNA-seq reference
data<a class="anchor" aria-label="anchor" href="#detect-signature-genes-for-each-cell-type-using-scrna-seq-reference-data"></a>
</h2>
<p>We introduce how to use CAESAR to detect signature genes form
scRNA-seq reference data. First, we calculate the co-embeddings.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MOB_scRNA</span> <span class="op">&lt;-</span> <span class="fu">ProFAST</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ProFAST/man/NCFM.html" class="external-link">NCFM</a></span><span class="op">(</span><span class="va">MOB_scRNA</span>, q <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size</span></span>
<span><span class="co">#&gt; 1.3 GiB</span></span>
<span><span class="co">#&gt; 2024-09-12 20:19:55.485497 : ***** Finish CoFAST, 0.137 mins elapsed.</span></span></code></pre></div>
<p>Then, we detect signature genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate cell-gene distance</span></span>
<span><span class="va">MOB_scRNA</span> <span class="op">&lt;-</span> <span class="fu">ProFAST</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ProFAST/man/pdistance.html" class="external-link">pdistance</a></span><span class="op">(</span><span class="va">MOB_scRNA</span>, reduction <span class="op">=</span> <span class="st">"ncfm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculate co-embedding distance...</span></span>
<span></span>
<span><span class="co"># identify signature genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">$</span><span class="va">CellType</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; EPL-IN     GC   M/TC   OSNs    PGC </span></span>
<span><span class="co">#&gt;    161   8614   1133   1200   1693</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">MOB_scRNA</span><span class="op">$</span><span class="va">CellType</span></span>
<span><span class="va">sg_sc_List</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find.sig.genes.html">find.sig.genes</a></span><span class="op">(</span><span class="va">MOB_scRNA</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">sg_sc_List</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 5</span></span>
<span><span class="co">#&gt;  $ EPL-IN:'data.frame':  13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 0.0672 0.0676 0.0677 0.0678 0.0678 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.0559 0.0807 0.0932 0.0621 0.0683 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.000791 0.001503 0.000712 0.001503 0.005696 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "EPL-IN" "EPL-IN" "EPL-IN" "EPL-IN" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Gpr101" "Col25a1" "Chrna2" "Mal2" ...</span></span>
<span><span class="co">#&gt;  $ GC    :'data.frame':  13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 0.061 0.061 0.061 0.0611 0.0611 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.334 0.419 0.329 0.341 0.573 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.152 0.225 0.173 0.204 0.343 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "GC" "GC" "GC" "GC" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Bcl11a" "Auts2" "Tnik" "Gabra2" ...</span></span>
<span><span class="co">#&gt;  $ M/TC  :'data.frame':  13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 0.063 0.063 0.063 0.0631 0.0631 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.685 0.518 0.145 0.336 0.703 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.0258 0.02211 0.00291 0.00471 0.03771 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "M/TC" "M/TC" "M/TC" "M/TC" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Cdhr1" "Nmb" "Dkkl1" "Tmem163" ...</span></span>
<span><span class="co">#&gt;  $ OSNs  :'data.frame':  13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 0.0575 0.0576 0.0576 0.0576 0.0576 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.1917 0.2125 0.2875 0.0575 0.4192 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.00862 0.01259 0.01983 0.00233 0.0325 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "OSNs" "OSNs" "OSNs" "OSNs" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Efcab10" "Tstd1" "Stoml3" "Sult1d1" ...</span></span>
<span><span class="co">#&gt;  $ PGC   :'data.frame':  13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 0.0536 0.0536 0.0537 0.0537 0.0537 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.0756 0.2936 0.2073 0.3438 0.0845 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.00765 0.0334 0.02782 0.03988 0.009 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "PGC" "PGC" "PGC" "PGC" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Igfbp3" "Cacnb2" "Cacna2d2" "AI593442" ...</span></span></code></pre></div>
<p>Finally, select marker genes for each cell type from the signature
gene list.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">marker</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/marker.select.html">marker.select</a></span><span class="op">(</span><span class="va">sg_sc_List</span>, overlap.max <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">marker</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`EPL-IN`</span></span>
<span><span class="co">#&gt;  [1] "Kit"      "Slc24a2"  "Cdh13"    "Slc7a14"  "Chl1"     "Lgi2"    </span></span>
<span><span class="co">#&gt;  [7] "Olfm3"    "Vgf"      "Shisa9"   "Resp18"   "Ache"     "Disp2"   </span></span>
<span><span class="co">#&gt; [13] "Thy1"     "Cntnap1"  "Tmem178"  "Clstn3"   "Enpp5"    "Slc35f1" </span></span>
<span><span class="co">#&gt; [19] "Sema3e"   "Endod1"   "Nell2"    "Nceh1"    "Slc6a17"  "Sorcs3"  </span></span>
<span><span class="co">#&gt; [25] "Gaa"      "Dner"     "Lgi3"     "Pcyox1"   "Tmem130"  "Syngr3"  </span></span>
<span><span class="co">#&gt; [31] "Kcnc2"    "Gns"      "Chga"     "Wnt5a"    "Megf8"    "Degs1"   </span></span>
<span><span class="co">#&gt; [37] "Lgmn"     "Atp9a"    "Cd200"    "Chgb"     "Ptprf"    "Spock2"  </span></span>
<span><span class="co">#&gt; [43] "Gas6"     "Ctsb"     "Syt13"    "Edil3"    "Omg"      "Slc35g2" </span></span>
<span><span class="co">#&gt; [49] "Gabra1"   "Smpd1"    "Tmem132b" "Lynx1"    "Syt4"     "Tmem38a" </span></span>
<span><span class="co">#&gt; [55] "Pcdh20"   "Hsd17b12" "Slc2a13"  "Plxna2"   "Ajap1"    "Bcap29"  </span></span>
<span><span class="co">#&gt; [61] "Cntn2"    "Mgat5b"   "Sez6l2"   "Syngr1"   "Nfe2l1"   "Slc9a6"  </span></span>
<span><span class="co">#&gt; [67] "Atp6ap1" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $GC</span></span>
<span><span class="co">#&gt;  [1] "Bcl11a"   "Auts2"    "Tnik"     "Gabra2"   "Celf2"    "Hivep2"  </span></span>
<span><span class="co">#&gt;  [7] "Sipa1l1"  "Rprml"    "Tmem191c" "Negr1"    "Ampd2"    "Plcb1"   </span></span>
<span><span class="co">#&gt; [13] "Ddx5"     "Il1rapl1" "Syne1"    "Fus"      "Grin2b"   "Ahi1"    </span></span>
<span><span class="co">#&gt; [19] "Cstf3"    "Cspg5"    "Ccnt2"    "Dcx"      "Qk"       "Pnn"     </span></span>
<span><span class="co">#&gt; [25] "Dnmt3a"   "Celf5"    "Celf3"    "Gria2"    "Zcchc9"   "Dlgap4"  </span></span>
<span><span class="co">#&gt; [31] "Thoc2"    "Luc7l3"   "Baiap2"   "Srsf11"   "Ogt"      "Tra2b"   </span></span>
<span><span class="co">#&gt; [37] "Rasa1"    "Abi2"     "Stxbp5l"  "Ddx17"    "Ralgps2"  "Ppp1cc"  </span></span>
<span><span class="co">#&gt; [43] "Actn4"    "Nudcd3"   "Meg3"     "Tia1"     "Prr7"     "Celf1"   </span></span>
<span><span class="co">#&gt; [49] "Sf3b1"    "Ank2"     "Arhgef2"  "Ewsr1"    "Nat14"    "Ppfia2"  </span></span>
<span><span class="co">#&gt; [55] "Uhrf2"    "Ube3a"    "Sag"      "Huwe1"    "Hnrnph1"  "Ccnl2"   </span></span>
<span><span class="co">#&gt; [61] "Dhx30"    "Golgb1"   "Ildr2"    "Ipo13"    "Leng8"    "Pbx3"    </span></span>
<span><span class="co">#&gt; [67] "Srsf2"   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`M/TC`</span></span>
<span><span class="co">#&gt;  [1] "Cdhr1"    "Nmb"      "Dkkl1"    "Tmem163"  "Slc17a7"  "Crim1"   </span></span>
<span><span class="co">#&gt;  [7] "Atp6ap1l" "Kitl"     "Thsd7a"   "Ms4a15"   "Chst15"   "Cbln1"   </span></span>
<span><span class="co">#&gt; [13] "Reln"     "Ptchd2"   "Spp1"     "Chst8"    "Sdc3"     "Kcna2"   </span></span>
<span><span class="co">#&gt; [19] "Crh"      "Sv2b"     "Nrn1"     "Sema3d"   "Scube1"   "Lrrtm1"  </span></span>
<span><span class="co">#&gt; [25] "Galnt14"  "Cntn4"    "Tspan9"   "Cbln4"    "Hapln4"   "Slc6a7"  </span></span>
<span><span class="co">#&gt; [31] "L1cam"    "Shisa3"   "Nptx1"    "Ndnf"     "Wbscr17"  "Lrrc55"  </span></span>
<span><span class="co">#&gt; [37] "Adcyap1"  "Igf1"     "Slc20a1"  "Lhfp"     "Sema7a"   "Lingo1"  </span></span>
<span><span class="co">#&gt; [43] "Adora1"   "Npr1"     "Slc38a3"  "Tgfa"     "Nell1"    "Otop2"   </span></span>
<span><span class="co">#&gt; [49] "Slc17a6"  "Mdga1"    "Eomes"    "Plcxd2"   "Adam11"   "Cyp4x1"  </span></span>
<span><span class="co">#&gt; [55] "Spock1"   "Tspan17"  "Hapln1"   "Ctgf"     "Atrnl1"   "Grm1"    </span></span>
<span><span class="co">#&gt; [61] "Nptx2"    "Cd302"    "Dhcr24"   "Abhd3"    "Scn2b"    "Grm7"    </span></span>
<span><span class="co">#&gt; [67] "Arhgdig" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $OSNs</span></span>
<span><span class="co">#&gt;  [1] "Efcab10"  "Tstd1"    "Stoml3"   "Omp"      "Pth2"     "Nrn1l"   </span></span>
<span><span class="co">#&gt;  [7] "Acbd7"    "Riiad1"   "Atf5"     "Spa17"    "Nme5"     "Fam213b" </span></span>
<span><span class="co">#&gt; [13] "Ctxn3"    "Trak2"    "Scrib"    "Jakmip1"  "Tshz2"    "Ppp1r14b"</span></span>
<span><span class="co">#&gt; [19] "Plekhb1"  "Gm20594"  "Mcf2l"    "Mycbp"    "Macrod1"  "Gng13"   </span></span>
<span><span class="co">#&gt; [25] "Tbc1d9b"  "S100a5"   "Atox1"    "Tesc"     "S100a13"  "Mt2"     </span></span>
<span><span class="co">#&gt; [31] "Tpt1"     "Anapc13"  "Pfdn2"    "Kif5b"    "Rps12"    "Rpl22"   </span></span>
<span><span class="co">#&gt; [37] "Uba52"    "Wdr89"    "Pdrg1"    "Rpl12"    "Ndufv3"   "Rpl30"   </span></span>
<span><span class="co">#&gt; [43] "Rplp2"    "Tomm7"    "Pfdn5"    "Rps20"    "Adipor1"  "Myeov2"  </span></span>
<span><span class="co">#&gt; [49] "Rpl23a"   "Rpl39"    "Rps24"    "S100b"    "Rpl28"    "Rps7"    </span></span>
<span><span class="co">#&gt; [55] "Rps9"     "Ost4"     "Rplp1"    "Fau"      "Rps2"     "Rpl10"   </span></span>
<span><span class="co">#&gt; [61] "Rps10"    "Ndufa1"   "Rpl36"    "Rps13"    "Rps21"    "Mt1"     </span></span>
<span><span class="co">#&gt; [67] "Sod1"    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $PGC</span></span>
<span><span class="co">#&gt;  [1] "Cacnb2"   "Cacna2d2" "AI593442" "Pde1c"    "Dcdc2a"   "Whrn"    </span></span>
<span><span class="co">#&gt;  [7] "Plvap"    "Rgs7"     "Gpr22"    "Nxph1"    "Syndig1l" "Cnr1"    </span></span>
<span><span class="co">#&gt; [13] "Pde5a"    "Asic4"    "Rgs10"    "Ttc28"    "Nppa"     "Sipa1l2" </span></span>
<span><span class="co">#&gt; [19] "Mtss1"    "Pard6g"   "Reps2"    "Mgll"     "Grem1"    "Kcnj12"  </span></span>
<span><span class="co">#&gt; [25] "Tle2"     "Car4"     "Dnajc12"  "Zfp385a"  "Pik3r3"   "Tmem176b"</span></span>
<span><span class="co">#&gt; [31] "Diras2"   "Inpp4b"   "Egf"      "Coprs"    "Msi1"     "Insm1"   </span></span>
<span><span class="co">#&gt; [37] "Gm11549"  "Cdkn1b"   "B3galt2"  "Enho"     "Pdzrn4"   "Plekhb2" </span></span>
<span><span class="co">#&gt; [43] "Samd14"   "Msi2"     "Mrps6"    "Magi1"    "Spon1"    "Crem"    </span></span>
<span><span class="co">#&gt; [49] "Kif22"    "Gng2"     "Apold1"   "Rnd3"     "Flywch2"  "Cyth1"   </span></span>
<span><span class="co">#&gt; [55] "Sbk1"     "Vstm2l"   "Pag1"     "Ppa1"     "Necab1"   "Fam49a"  </span></span>
<span><span class="co">#&gt; [61] "Foxn3"    "Dbp"      "Tmem108"  "Tmem169"  "Fabp3"    "Nap1l5"  </span></span>
<span><span class="co">#&gt; [67] "Prmt2"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="annotate-the-mob-st-data-using-caesar-and-marker-genes-from-scrna-seq-reference-data">Annotate the MOB ST data using CAESAR and marker genes from
scRNA-seq reference data<a class="anchor" aria-label="anchor" href="#annotate-the-mob-st-data-using-caesar-and-marker-genes-from-scrna-seq-reference-data"></a>
</h2>
<p>Similarly, we first calculate co-embeddings for MOB ST dataset. The
difference is that spatial transcriptome data has spatial coordinates
information, so we can obtain spatial aware co-embeddings.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the spatial coordinates</span></span>
<span><span class="va">pos</span> <span class="op">&lt;-</span> <span class="va">MOB_ST</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    x      y</span></span>
<span><span class="co">#&gt; 16.918x16.996 16.918 16.996</span></span>
<span><span class="co">#&gt; 18.017x17.034 18.017 17.034</span></span>
<span><span class="co">#&gt; 20.075x17.059 20.075 17.059</span></span>
<span><span class="co">#&gt; 18.979x17.065 18.979 17.065</span></span>
<span><span class="co">#&gt; 21.937x16.967 21.937 16.967</span></span>
<span><span class="co">#&gt; 20.899x17.015 20.899 17.015</span></span>
<span></span>
<span><span class="va">MOB_ST</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CAESAR.coembedding.html">CAESAR.coembedding</a></span><span class="op">(</span><span class="va">MOB_ST</span>, <span class="va">pos</span>, reduction.name <span class="op">=</span> <span class="st">"caesar"</span>, q <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="co">#&gt; The spatial cooridnates are 2 dimensions</span></span>
<span><span class="co">#&gt; Find the adjacency matrix by bisection method...</span></span>
<span><span class="co">#&gt; Current radius is 200.5</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 2</span></span>
<span><span class="co">#&gt; Current radius is 100.75</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 279</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 279</span></span>
<span><span class="co">#&gt; Current radius is 25.94</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 279</span></span>
<span><span class="co">#&gt; Current radius is 13.47</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 279</span></span>
<span><span class="co">#&gt; Current radius is 7.23</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 233</span></span>
<span><span class="co">#&gt; Current radius is 4.12</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 105.5</span></span>
<span><span class="co">#&gt; Current radius is 2.56</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 43</span></span>
<span><span class="co">#&gt; Current radius is 1.78</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 19</span></span>
<span><span class="co">#&gt; Current radius is 1.78</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 8</span></span>
<span><span class="co">#&gt; Finish variable intialization </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 2, loglik= 93961.618824, dloglik=1.000044 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 3, loglik= 95904.848879, dloglik=0.020681 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 4, loglik= 96392.301448, dloglik=0.005083 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 5, loglik= 96683.214511, dloglik=0.003018 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 6, loglik= 96874.024782, dloglik=0.001974 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 7, loglik= 97008.159784, dloglik=0.001385 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 8, loglik= 97106.416775, dloglik=0.001013 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 9, loglik= 97180.663023, dloglik=0.000765 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 10, loglik= 97238.089763, dloglik=0.000591 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 11, loglik= 97283.339881, dloglik=0.000465 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 12, loglik= 97319.556554, dloglik=0.000372 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 13, loglik= 97348.931983, dloglik=0.000302 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 14, loglik= 97373.021477, dloglik=0.000247 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 15, loglik= 97392.947057, dloglik=0.000205 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 16, loglik= 97409.538654, dloglik=0.000170 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 17, loglik= 97423.430702, dloglik=0.000143 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 18, loglik= 97435.124354, dloglik=0.000120 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 19, loglik= 97445.025103, dloglik=0.000102 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 20, loglik= 97453.464916, dloglik=0.000087 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 21, loglik= 97460.715878, dloglik=0.000074 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 22, loglik= 97466.999772, dloglik=0.000064 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 23, loglik= 97472.495863, dloglik=0.000056 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 24, loglik= 97477.347771, dloglik=0.000050 </span></span>
<span><span class="co">#&gt; Satrt ICM and E-step! </span></span>
<span><span class="co">#&gt; Finish ICM and E-step! </span></span>
<span><span class="co">#&gt; iter = 25, loglik= 97481.669671, dloglik=0.000044</span></span>
<span><span class="co">#&gt; 2024-09-12 20:20:05.539317 : ***** Finish calculate CAESAR embedding, 0.035 mins elapsed.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 13895 features across 280 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (13895 features, 508 variable features)</span></span>
<span><span class="co">#&gt;  1 dimensional reduction calculated: caesar</span></span></code></pre></div>
<p>Subsequently, the CAESAR co-embeddings and marker genes from
scRNA-seq reference data are used to annotate the MOB ST data.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># convert marker list to marker frequency matrix</span></span>
<span><span class="va">marker.freq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/markerList2mat.html">markerList2mat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">marker</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform annotation using CAESAR and save results to Seurat object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "orig.ident"        "nCount_RNA"        "nFeature_RNA"     </span></span>
<span><span class="co">#&gt; [4] "x"                 "y"                 "manual_annotation"</span></span>
<span></span>
<span><span class="va">MOB_ST</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CAESAR.annotation.html">CAESAR.annotation</a></span><span class="op">(</span><span class="va">MOB_ST</span>, <span class="va">marker.freq</span>, reduction.name <span class="op">=</span> <span class="st">"caesar"</span>, add.to.meta <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculate co-embedding distance...</span></span>
<span><span class="co">#&gt; Warning: package 'purrr' was built under R version 4.3.3</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "orig.ident"        "nCount_RNA"        "nFeature_RNA"     </span></span>
<span><span class="co">#&gt;  [4] "x"                 "y"                 "manual_annotation"</span></span>
<span><span class="co">#&gt;  [7] "CAESAR"            "CAESARunasg"       "CAESARconf"       </span></span>
<span><span class="co">#&gt; [10] "dist_EPL.IN"       "dist_GC"           "dist_M.TC"        </span></span>
<span><span class="co">#&gt; [13] "dist_OSNs"         "dist_PGC"          "prob_EPL.IN"      </span></span>
<span><span class="co">#&gt; [16] "prob_GC"           "prob_M.TC"         "prob_OSNs"        </span></span>
<span><span class="co">#&gt; [19] "prob_PGC"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="downstream-analysis">Downstream analysis<a class="anchor" aria-label="anchor" href="#downstream-analysis"></a>
</h2>
<p>In the following, we visualize the CAESAR annotation results.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set up colors</span></span>
<span><span class="va">cols_manual</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"#4374A5"</span>, <span class="st">"#FCDDDE"</span>, <span class="st">"#2AB67F"</span>, <span class="st">"#F08A21"</span>, <span class="st">"#737373"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"GCL"</span>, <span class="st">"MCL"</span>, <span class="st">"ONL"</span>, <span class="st">"GL"</span>, <span class="st">"Unknown"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">celltypes_manual</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GCL"</span>, <span class="st">"MCL"</span>, <span class="st">"ONL"</span>, <span class="st">"GL"</span>, <span class="st">"Unknown"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"#4374A5"</span>, <span class="st">"#FCDDDE"</span>, <span class="st">"#2AB673"</span>, <span class="st">"#F08A21"</span>, <span class="st">"#E04D50"</span>, <span class="st">"#737373"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"GC"</span>, <span class="st">"M/TC"</span>, <span class="st">"OSNs"</span>, <span class="st">"PGC"</span>, <span class="st">"EPL-IN"</span>, <span class="st">"unassigned"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GC"</span>, <span class="st">"M/TC"</span>, <span class="st">"OSNs"</span>, <span class="st">"PGC"</span>, <span class="st">"EPL-IN"</span>, <span class="st">"unassigned"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pos"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">MOB_ST</span><span class="op">@</span><span class="va">reductions</span><span class="op">[[</span><span class="st">"pos"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateDimReducObject.html" class="external-link">CreateDimReducObject</a></span><span class="op">(</span></span>
<span>    embeddings <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span>,</span>
<span>    key <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pos"</span>, <span class="st">"_"</span><span class="op">)</span>, assay <span class="op">=</span> <span class="st">"RNA"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>First, we visualize the manual annotation.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">$</span><span class="va">manual_annotation</span>, levels <span class="op">=</span> <span class="va">celltypes_manual</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">MOB_ST</span>, reduction <span class="op">=</span> <span class="st">"pos"</span>, cols <span class="op">=</span> <span class="va">cols_manual</span>, pt.size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<p><img src="STMOB_files/figure-html/unnamed-chunk-12-1.png" width="844.8"></p>
<p>Then, we visualize the CAESAR annotation without account for
‘unassigned’.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">$</span><span class="va">CAESAR</span>, levels <span class="op">=</span> <span class="va">celltypes</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">MOB_ST</span>, reduction <span class="op">=</span> <span class="st">"pos"</span>, cols <span class="op">=</span> <span class="va">cols</span>, pt.size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<p><img src="STMOB_files/figure-html/unnamed-chunk-13-1.png" width="844.8"></p>
<p>And visualize the CAESAR annotation account for ‘unassigned’.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">$</span><span class="va">CAESARunasg</span>, levels <span class="op">=</span> <span class="va">celltypes</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">MOB_ST</span>, reduction <span class="op">=</span> <span class="st">"pos"</span>, cols <span class="op">=</span> <span class="va">cols</span>, pt.size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<p><img src="STMOB_files/figure-html/unnamed-chunk-14-1.png" width="844.8"></p>
<p>The confidence level of the CAESAR annotation can be visualized
by</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span></span>
<span>    <span class="va">MOB_ST</span>,</span>
<span>    reduction <span class="op">=</span> <span class="st">"pos"</span>, features <span class="op">=</span> <span class="st">"CAESARconf"</span>, pt.size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>    cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"lightgrey"</span><span class="op">)</span>, min.cutoff <span class="op">=</span> <span class="fl">0.0</span>, max.cutoff <span class="op">=</span> <span class="fl">1.0</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="STMOB_files/figure-html/unnamed-chunk-15-1.png" width="844.8"></p>
<p>CAESAR provides the cell mixing proportion for each cell type, which
can be visualized by</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">caesar_prob</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span><span class="op">[</span><span class="fl">15</span><span class="op">:</span><span class="fl">19</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">caesar_prob</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "prob_EPL.IN" "prob_GC"     "prob_M.TC"   "prob_OSNs"   "prob_PGC"</span></span>
<span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">caesar_prob</span>, <span class="kw">function</span><span class="op">(</span><span class="va">feature</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">MOB_ST</span>, features <span class="op">=</span> <span class="va">feature</span>, reduction <span class="op">=</span> <span class="st">"pos"</span>, pt.size <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span></span>
<span>            colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#f6eff7"</span>, <span class="st">"#feebe2"</span>, <span class="st">"#f768a1"</span>, <span class="st">"#7a0177"</span>, <span class="st">"#6e016b"</span><span class="op">)</span>,</span>
<span>            values <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/rescale.html" class="external-link">rescale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.0</span>, <span class="fl">0.125</span>, <span class="fl">0.25</span>, <span class="fl">0.375</span>, <span class="fl">0.50</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.0</span>, <span class="fl">0.50</span><span class="op">)</span></span>
<span>        <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="va">feature</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span></span>
<span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">plots</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="STMOB_files/figure-html/unnamed-chunk-16-1.png" width="844.8"></p>
<p>The annotation accuracy is calculated by</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc_st</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">manual_annotation</span>, <span class="va">pred</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">manual_annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">manual_annotation</span><span class="op">)</span></span>
<span>    <span class="va">pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">pred</span><span class="op">)</span></span>
<span>    <span class="va">manual_annotation</span><span class="op">[</span><span class="va">manual_annotation</span> <span class="op">==</span> <span class="st">"GCL"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"GC"</span></span>
<span>    <span class="va">manual_annotation</span><span class="op">[</span><span class="va">manual_annotation</span> <span class="op">==</span> <span class="st">"MCL"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"M/TC"</span></span>
<span>    <span class="va">manual_annotation</span><span class="op">[</span><span class="va">manual_annotation</span> <span class="op">==</span> <span class="st">"ONL"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"OSNs"</span></span>
<span>    <span class="va">manual_annotation</span><span class="op">[</span><span class="va">manual_annotation</span> <span class="op">==</span> <span class="st">"GL"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"PGC"</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">manual_annotation</span> <span class="op">==</span> <span class="va">pred</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>    <span class="st">"The ACC of CAESAR annotation is "</span>,</span>
<span>    <span class="fu">acc_st</span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">$</span><span class="va">manual_annotation</span>, <span class="va">MOB_ST</span><span class="op">$</span><span class="va">CAESARunasg</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "The ACC of CAESAR annotation is 0.717857142857143"</span></span></code></pre></div>
<p>Next, we detect and visualize the signature genes for each cell
type.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">$</span><span class="va">CAESARunasg</span>, <span class="va">celltypes</span><span class="op">)</span></span>
<span><span class="va">sg_List</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find.sig.genes.html">find.sig.genes</a></span><span class="op">(</span><span class="va">MOB_ST</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">sg_List</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 4</span></span>
<span><span class="co">#&gt;  $ GC  :'data.frame':    13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 5.49 5.49 5.52 5.53 5.56 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.344 0.422 0.322 0.322 0.256 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.0842 0.0947 0.0579 0.0421 0.0421 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "GC" "GC" "GC" "GC" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Cdkn1a" "Rasgrp2" "Kcnh5" "Kcnh1" ...</span></span>
<span><span class="co">#&gt;  $ M/TC:'data.frame':    13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 6.69 6.74 6.74 6.75 6.75 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.3494 0.0964 0.6627 0.7711 0.3012 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.1066 0.0152 0.2792 0.3147 0.1015 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "M/TC" "M/TC" "M/TC" "M/TC" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Raver2" "Slc16a10" "Zfp804a" "Nefl" ...</span></span>
<span><span class="co">#&gt;  $ OSNs:'data.frame':    13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 9.69 9.72 9.73 9.73 9.73 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.836 0.448 0.612 0.701 0.493 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.2582 0.0892 0.1643 0.1315 0.1174 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "OSNs" "OSNs" "OSNs" "OSNs" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Hmgcs2" "Alx4" "Afap1l2" "Igfbp4" ...</span></span>
<span><span class="co">#&gt;  $ PGC :'data.frame':    13895 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ distance        : num [1:13895] 7.71 7.74 7.74 7.77 7.77 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop       : num [1:13895] 0.55 0.825 0.6 0.35 0.5 0.075 0.375 0.2 0.1 0.85 ...</span></span>
<span><span class="co">#&gt;   ..$ expr.prop.others: num [1:13895] 0.183 0.412 0.267 0.188 0.192 ...</span></span>
<span><span class="co">#&gt;   ..$ label           : chr [1:13895] "PGC" "PGC" "PGC" "PGC" ...</span></span>
<span><span class="co">#&gt;   ..$ gene            : chr [1:13895] "Cnr1" "AI593442" "Shisa6" "Kcnj12" ...</span></span></code></pre></div>
<p>We visualize the top three signature genes by a dot plot.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># obtain the top three signature genes</span></span>
<span><span class="va">celltypes_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sg_List</span><span class="op">)</span>, <span class="st">"unassigned"</span><span class="op">)</span></span>
<span><span class="va">top3sgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Intsg.html">Intsg</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">sg_List</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span><span class="op">[</span><span class="va">celltypes_plot</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">top3sgs</span><span class="op">)</span></span>
<span><span class="co">#&gt; $GC</span></span>
<span><span class="co">#&gt; [1] "Cdkn1a"  "Rasgrp2" "Kcnh5"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`M/TC`</span></span>
<span><span class="co">#&gt; [1] "Raver2"  "Zfp804a" "Nefl"   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $OSNs</span></span>
<span><span class="co">#&gt; [1] "Hmgcs2"  "Alx4"    "Afap1l2"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $PGC</span></span>
<span><span class="co">#&gt; [1] "Cnr1"     "AI593442" "Shisa6"</span></span>
<span></span>
<span><span class="va">sg_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">top3sgs</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span></span>
<span>    <span class="va">MOB_ST</span>,</span>
<span>    idents <span class="op">=</span> <span class="va">celltypes_plot</span>, col.min <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, col.max <span class="op">=</span> <span class="fl">2</span>, dot.scale <span class="op">=</span> <span class="fl">7</span>,</span>
<span>    features <span class="op">=</span> <span class="va">sg_features</span>, scale.min <span class="op">=</span> <span class="fl">0</span>, scale.max <span class="op">=</span> <span class="fl">30</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span>, angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Scaling data with a low number of groups may produce misleading</span></span>
<span><span class="co">#&gt; results</span></span></code></pre></div>
<p><img src="STMOB_files/figure-html/unnamed-chunk-19-1.png" width="844.8"></p>
<p>Next, we calculate the UMAP projections of co-embeddings of cells and
the selected signature genes.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate coumap</span></span>
<span><span class="va">MOB_ST</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CoUMAP.html">CoUMAP</a></span><span class="op">(</span></span>
<span>    <span class="va">MOB_ST</span>, reduction <span class="op">=</span> <span class="st">"caesar"</span>, reduction.name <span class="op">=</span> <span class="st">"caesarUMAP"</span>,</span>
<span>    gene.set <span class="op">=</span> <span class="va">sg_features</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">df_gene_label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">top3sgs</span><span class="op">)</span>,</span>
<span>    label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">top3sgs</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/CoUMAP.plot.html">CoUMAP.plot</a></span><span class="op">(</span></span>
<span>    <span class="va">MOB_ST</span>, reduction <span class="op">=</span> <span class="st">"caesarUMAP"</span>, gene_txtdata <span class="op">=</span> <span class="va">df_gene_label</span>,</span>
<span>    cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span> <span class="op">=</span> <span class="st">"#000000"</span>, <span class="va">cols</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="STMOB_files/figure-html/unnamed-chunk-20-1.png" width="844.8"></p>
<details><summary><strong>Session Info</strong>
</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 22631)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Chinese (Simplified)_China.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Chinese (Simplified)_China.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Chinese (Simplified)_China.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                               </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Chinese (Simplified)_China.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Shanghai</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] purrr_1.0.2        ggplot2_3.5.1      ProFAST_1.4        gtools_3.9.5      </span></span>
<span><span class="co">#&gt; [5] Seurat_4.4.0       CAESAR.Suite_0.1.0 SeuratObject_4.1.4 sp_2.1-4          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] fs_1.6.4                    matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt;   [3] spatstat.sparse_3.1-0       bitops_1.0-8               </span></span>
<span><span class="co">#&gt;   [5] httr_1.4.7                  RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;   [7] tools_4.3.1                 sctransform_0.4.1          </span></span>
<span><span class="co">#&gt;   [9] backports_1.5.0             utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [11] R6_2.5.1                    lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [13] uwot_0.2.2                  withr_3.0.1                </span></span>
<span><span class="co">#&gt;  [15] prettyunits_1.2.0           gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [17] progressr_0.14.0            cli_3.6.3                  </span></span>
<span><span class="co">#&gt;  [19] Biobase_2.62.0              textshaping_0.4.0          </span></span>
<span><span class="co">#&gt;  [21] spatstat.explore_3.3-2      labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [23] sass_0.4.9                  mvtnorm_1.2-6              </span></span>
<span><span class="co">#&gt;  [25] spatstat.data_3.1-2         proxy_0.4-27               </span></span>
<span><span class="co">#&gt;  [27] ggridges_0.5.6              pbapply_1.7-2              </span></span>
<span><span class="co">#&gt;  [29] pkgdown_2.0.6               systemfonts_1.1.0          </span></span>
<span><span class="co">#&gt;  [31] harmony_1.2.1               scater_1.30.1              </span></span>
<span><span class="co">#&gt;  [33] parallelly_1.38.0           readxl_1.4.3               </span></span>
<span><span class="co">#&gt;  [35] rstudioapi_0.16.0           RSQLite_2.3.7              </span></span>
<span><span class="co">#&gt;  [37] FNN_1.1.4                   generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [39] ica_1.0-3                   spatstat.random_3.3-1      </span></span>
<span><span class="co">#&gt;  [41] car_3.1-2                   dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [43] Matrix_1.6-1.1              ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt;  [45] fansi_1.0.6                 DescTools_0.99.56          </span></span>
<span><span class="co">#&gt;  [47] S4Vectors_0.40.2            abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [49] lifecycle_1.0.4             yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [51] CompQuadForm_1.4.3          carData_3.0-5              </span></span>
<span><span class="co">#&gt;  [53] SummarizedExperiment_1.32.0 SparseArray_1.2.4          </span></span>
<span><span class="co">#&gt;  [55] BiocFileCache_2.10.2        Rtsne_0.17                 </span></span>
<span><span class="co">#&gt;  [57] grid_4.3.1                  blob_1.2.4                 </span></span>
<span><span class="co">#&gt;  [59] promises_1.3.0              crayon_1.5.3               </span></span>
<span><span class="co">#&gt;  [61] GiRaF_1.0.1                 miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [63] lattice_0.21-8              beachmat_2.18.1            </span></span>
<span><span class="co">#&gt;  [65] cowplot_1.1.3               KEGGREST_1.42.0            </span></span>
<span><span class="co">#&gt;  [67] pillar_1.9.0                knitr_1.48                 </span></span>
<span><span class="co">#&gt;  [69] GenomicRanges_1.54.1        boot_1.3-28.1              </span></span>
<span><span class="co">#&gt;  [71] gld_2.6.6                   future.apply_1.11.2        </span></span>
<span><span class="co">#&gt;  [73] codetools_0.2-19            leiden_0.4.3.1             </span></span>
<span><span class="co">#&gt;  [75] glue_1.7.0                  spatstat.univar_3.0-0      </span></span>
<span><span class="co">#&gt;  [77] data.table_1.16.0           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [79] png_0.1-8                   spam_2.10-0                </span></span>
<span><span class="co">#&gt;  [81] org.Mm.eg.db_3.18.0         cellranger_1.1.0           </span></span>
<span><span class="co">#&gt;  [83] gtable_0.3.5                cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [85] xfun_0.47                   S4Arrays_1.2.1             </span></span>
<span><span class="co">#&gt;  [87] mime_0.12                   survival_3.5-5             </span></span>
<span><span class="co">#&gt;  [89] SingleCellExperiment_1.24.0 fitdistrplus_1.2-1         </span></span>
<span><span class="co">#&gt;  [91] ROCR_1.0-11                 nlme_3.1-162               </span></span>
<span><span class="co">#&gt;  [93] bit64_4.0.5                 filelock_1.0.3             </span></span>
<span><span class="co">#&gt;  [95] progress_1.2.3              RcppAnnoy_0.0.22           </span></span>
<span><span class="co">#&gt;  [97] GenomeInfoDb_1.38.8         bslib_0.8.0                </span></span>
<span><span class="co">#&gt;  [99] irlba_2.3.5.1               vipor_0.4.7                </span></span>
<span><span class="co">#&gt; [101] KernSmooth_2.23-21          colorspace_2.1-1           </span></span>
<span><span class="co">#&gt; [103] BiocGenerics_0.48.1         DBI_1.2.3                  </span></span>
<span><span class="co">#&gt; [105] ade4_1.7-22                 Exact_3.3                  </span></span>
<span><span class="co">#&gt; [107] tidyselect_1.2.1            DR.SC_3.4                  </span></span>
<span><span class="co">#&gt; [109] curl_5.2.2                  bit_4.0.5                  </span></span>
<span><span class="co">#&gt; [111] compiler_4.3.1              BiocNeighbors_1.20.2       </span></span>
<span><span class="co">#&gt; [113] expm_1.0-0                  xml2_1.3.6                 </span></span>
<span><span class="co">#&gt; [115] desc_1.4.3                  DelayedArray_0.28.0        </span></span>
<span><span class="co">#&gt; [117] plotly_4.10.4               scales_1.3.0               </span></span>
<span><span class="co">#&gt; [119] lmtest_0.9-40               rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt; [121] stringr_1.5.1               digest_0.6.37              </span></span>
<span><span class="co">#&gt; [123] goftest_1.2-3               spatstat.utils_3.1-0       </span></span>
<span><span class="co">#&gt; [125] rmarkdown_2.28              XVector_0.42.0             </span></span>
<span><span class="co">#&gt; [127] htmltools_0.5.8.1           pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [129] sparseMatrixStats_1.14.0    MatrixGenerics_1.14.0      </span></span>
<span><span class="co">#&gt; [131] highr_0.11                  dbplyr_2.5.0               </span></span>
<span><span class="co">#&gt; [133] fastmap_1.2.0               rlang_1.1.4                </span></span>
<span><span class="co">#&gt; [135] htmlwidgets_1.6.4           ggthemes_5.1.0             </span></span>
<span><span class="co">#&gt; [137] shiny_1.9.1                 DelayedMatrixStats_1.24.0  </span></span>
<span><span class="co">#&gt; [139] farver_2.1.2                jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [141] zoo_1.8-12                  jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt; [143] BiocParallel_1.36.0         mclust_6.1.1               </span></span>
<span><span class="co">#&gt; [145] BiocSingular_1.18.0         RCurl_1.98-1.16            </span></span>
<span><span class="co">#&gt; [147] magrittr_2.0.3              scuttle_1.12.0             </span></span>
<span><span class="co">#&gt; [149] GenomeInfoDbData_1.2.11     dotCall64_1.1-1            </span></span>
<span><span class="co">#&gt; [151] patchwork_1.2.0             munsell_0.5.1              </span></span>
<span><span class="co">#&gt; [153] Rcpp_1.0.13                 viridis_0.6.5              </span></span>
<span><span class="co">#&gt; [155] reticulate_1.38.0           furrr_0.3.1                </span></span>
<span><span class="co">#&gt; [157] stringi_1.8.4               rootSolve_1.8.2.4          </span></span>
<span><span class="co">#&gt; [159] zlibbioc_1.48.2             MASS_7.3-60                </span></span>
<span><span class="co">#&gt; [161] org.Hs.eg.db_3.18.0         plyr_1.8.9                 </span></span>
<span><span class="co">#&gt; [163] parallel_4.3.1              PRECAST_1.6.5              </span></span>
<span><span class="co">#&gt; [165] listenv_0.9.1               ggrepel_0.9.5              </span></span>
<span><span class="co">#&gt; [167] lmom_3.0                    deldir_2.0-4               </span></span>
<span><span class="co">#&gt; [169] Biostrings_2.70.3           splines_4.3.1              </span></span>
<span><span class="co">#&gt; [171] tensor_1.5                  hms_1.1.3                  </span></span>
<span><span class="co">#&gt; [173] igraph_2.0.3                ggpubr_0.6.0               </span></span>
<span><span class="co">#&gt; [175] spatstat.geom_3.3-2         ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt; [177] reshape2_1.4.4              biomaRt_2.58.2             </span></span>
<span><span class="co">#&gt; [179] stats4_4.3.1                ScaledMatrix_1.10.0        </span></span>
<span><span class="co">#&gt; [181] XML_3.99-0.17               evaluate_0.24.0            </span></span>
<span><span class="co">#&gt; [183] httpuv_1.6.15               RANN_2.6.2                 </span></span>
<span><span class="co">#&gt; [185] tidyr_1.3.1                 polyclip_1.10-7            </span></span>
<span><span class="co">#&gt; [187] future_1.34.0               scattermore_1.2            </span></span>
<span><span class="co">#&gt; [189] rsvd_1.0.5                  broom_1.0.6                </span></span>
<span><span class="co">#&gt; [191] xtable_1.8-4                e1071_1.7-14               </span></span>
<span><span class="co">#&gt; [193] rstatix_0.7.2               later_1.3.2                </span></span>
<span><span class="co">#&gt; [195] viridisLite_0.4.2           class_7.3-22               </span></span>
<span><span class="co">#&gt; [197] ragg_1.3.2                  tibble_3.2.1               </span></span>
<span><span class="co">#&gt; [199] memoise_2.0.1               beeswarm_0.4.0             </span></span>
<span><span class="co">#&gt; [201] AnnotationDbi_1.64.1        IRanges_2.36.0             </span></span>
<span><span class="co">#&gt; [203] cluster_2.1.4               globals_0.16.3</span></span></code></pre></div>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Xiao Zhang, Wei Liu, Jin Liu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
