<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Xenium BC data analysis • CAESAR.Suite</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Xenium BC data analysis">
<meta property="og:description" content="CAESAR.Suite">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CAESAR.Suite</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/STMOB.html">ST MOB data analysis</a>
    </li>
    <li>
      <a href="../articles/XeniumBC.html">Xenium BC data analysis</a>
    </li>
    <li>
      <a href="../articles/XeniumBCEIF.html">Xenium BC extract image features</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/XiaoZhangryy/CAESAR.Suite/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Xenium BC data analysis</h1>
                        <h4 data-toc-skip class="author">Xiao Zhang</h4>
            
            <h4 data-toc-skip class="date">2024-09-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/XiaoZhangryy/CAESAR.Suite/blob/HEAD/vignettes/XeniumBC.Rmd" class="external-link"><code>vignettes/XeniumBC.Rmd</code></a></small>
      <div class="hidden name"><code>XeniumBC.Rmd</code></div>

    </div>

    
    
<p>This vignette introduces the CAESAR.Suite workflow for the analysis
of Xenium BC spatial transcriptomics dataset. In this vignette, the
workflow of CAESAR.Suite consists of five steps</p>
<ul>
<li>Reference dataset and target dataset preprocessing</li>
<li>Detect signature genes as cell type markers from scRNA-seq reference
datasets</li>
<li>Annotate Xenium BC data using CAESAR</li>
<li>Enrichment analysis for Xenium BC data using CAESAR</li>
<li>Downstream analysis (i.e. , signature gene analysis, visualization
of cell types and coembeddings)</li>
</ul>
<div class="section level2">
<h2 id="load-and-quality-control-both-reference-and-target-data">Load and quality control both reference and target data<a class="anchor" aria-label="anchor" href="#load-and-quality-control-both-reference-and-target-data"></a>
</h2>
<p>We demonstrate the application of CAESAR.Suite to Xenium data. In
this vignette, the input data includes: <code>BC_scRNAList</code> — two
scRNA-seq reference datasets, each with 3,000 cells;
<code>BC_XeniumList</code> — two Xenium target datasets, each with 3,000
cells; and <code>BC_feature_imgList</code> — two feature matrices
containing histology image information from the Xenium target datasets.
For more details with respect to histology image feature extraction, see
<a href="%22https://xiaozhangryy.github.io/CAESAR.Suite/articles/XeniumBCEIF.html" class="external-link">vignette</a>.
The genes of scRNA-seq reference datasets and Xenium target datasets are
aligned. The package can be loaded with the command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="co"># set a random seed for reproducibility.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/XiaoZhangryy/CAESAR.Suite" class="external-link">CAESAR.Suite</a></span><span class="op">)</span> <span class="co"># load the package of CAESAR.Suite method</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Attaching SeuratObject</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/feiyoung/ProFAST" class="external-link">ProFAST</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'ProFAST' was built under R version 4.3.3</span></span>
<span><span class="co">#&gt; Loading required package: gtools</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/" class="external-link">msigdbr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'msigdbr' was built under R version 4.3.3</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<p>Those data can be downloaded and load to the current working path by
the following command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/XiaoZhangryy/CAESAR.Suite/blob/master/vignettes_data/BC_scRNAList.rda?raw=true"</span></span>
<span><span class="va">BC_scRNAList_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"BC_scRNAList.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="va">BC_scRNAList_file</span>, mode<span class="op">=</span><span class="st">'wb'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">BC_scRNAList_file</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">BC_scRNAList</span><span class="op">)</span></span>
<span><span class="co">#&gt; $Ref1</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 7023 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 8609 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 0 variable features)</span></span>
<span></span>
<span><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/XiaoZhangryy/CAESAR.Suite/blob/master/vignettes_data/BC_XeniumList.rda?raw=true"</span></span>
<span><span class="va">BC_XeniumList_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"BC_XeniumList.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="va">BC_XeniumList_file</span>, mode<span class="op">=</span><span class="st">'wb'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">BC_XeniumList_file</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">BC_XeniumList</span><span class="op">)</span></span>
<span><span class="co">#&gt; $BC1</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 3000 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $BC2</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 3000 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 0 variable features)</span></span>
<span></span>
<span><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/XiaoZhangryy/CAESAR.Suite/blob/master/vignettes_data/BC_feature_imgList.rda?raw=true"</span></span>
<span><span class="va">BC_feature_imgList_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"BC_feature_imgList.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="va">BC_feature_imgList_file</span>, mode<span class="op">=</span><span class="st">'wb'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">BC_feature_imgList_file</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">BC_feature_imgList</span>, <span class="va">dim</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;       BC1  BC2</span></span>
<span><span class="co">#&gt; [1,] 3000 3000</span></span>
<span><span class="co">#&gt; [2,]  576  576</span></span></code></pre></div>
<p>Users can perform appropriate quality control on the reference
dataset and target datasets. Genes expressed in less than one cell are
required to remove to avoid unknown errors. Other quality control steps
can be set by the user according to the data quality. Here, since
scRNA-seq reference datasets and Xenium target datasets had been
aligned, we do not perform quality control.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># BC_scRNAList &lt;- lapply(BC_scRNAList,  function(seu) {</span></span>
<span><span class="co">#     CreateSeuratObject(</span></span>
<span><span class="co">#         counts = seu@assays$RNA@counts,</span></span>
<span><span class="co">#         meta.data = seu@meta.data,</span></span>
<span><span class="co">#         min.features = 5,</span></span>
<span><span class="co">#         min.cells = 1</span></span>
<span><span class="co">#     )</span></span>
<span><span class="co"># })</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># print(BC_scRNAList)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># BC_XeniumList &lt;- lapply(BC_XeniumList,  function(seu) {</span></span>
<span><span class="co">#     CreateSeuratObject(</span></span>
<span><span class="co">#         counts = seu@assays$RNA@counts,</span></span>
<span><span class="co">#         meta.data = seu@meta.data,</span></span>
<span><span class="co">#         min.features = 5,</span></span>
<span><span class="co">#         min.cells = 1</span></span>
<span><span class="co">#     )</span></span>
<span><span class="co"># })</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># print(BC_XeniumList)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># BC_feature_imgList &lt;- lapply(1:2, function(i) {</span></span>
<span><span class="co">#     BC_feature_imgList[[i]][colnames(BC_XeniumList[[i]]), ]</span></span>
<span><span class="co"># })</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocessing-and-align-reference-and-target-data">Preprocessing and align reference and target data<a class="anchor" aria-label="anchor" href="#preprocessing-and-align-reference-and-target-data"></a>
</h2>
<p>First, we normalize the data and select the variable genes. We align
genes and variable genes of reference and target data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># align genes</span></span>
<span><span class="va">common_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">intersect</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_scRNAList</span>, <span class="va">rownames</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="va">rownames</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">common_genes</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 302</span></span>
<span></span>
<span><span class="co"># all common genes are used as variable genes, as only around 300 genes here</span></span>
<span><span class="va">BC_scRNAList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_scRNAList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">seu</span> <span class="op">&lt;-</span> <span class="va">seu</span><span class="op">[</span><span class="va">common_genes</span>, <span class="op">]</span></span>
<span>    <span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">common_genes</span></span>
<span>    <span class="va">seu</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">BC_XeniumList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">seu</span> <span class="op">&lt;-</span> <span class="va">seu</span><span class="op">[</span><span class="va">common_genes</span>, <span class="op">]</span></span>
<span>    <span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">common_genes</span></span>
<span>    <span class="va">seu</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">BC_scRNAList</span><span class="op">)</span></span>
<span><span class="co">#&gt; $Ref1</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 7023 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 302 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 8609 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 302 variable features)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">BC_XeniumList</span><span class="op">)</span></span>
<span><span class="co">#&gt; $BC1</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 3000 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 302 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $BC2</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 3000 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 302 variable features)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="detect-signature-genes-for-each-cell-type-using-scrna-seq-reference-data">Detect signature genes for each cell type using scRNA-seq reference
data<a class="anchor" aria-label="anchor" href="#detect-signature-genes-for-each-cell-type-using-scrna-seq-reference-data"></a>
</h2>
<p>We introduce how to use CAESAR to detect signature genes form
scRNA-seq reference data. First, we calculate the co-embeddings.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BC_scRNAList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_scRNAList</span>, <span class="fu">ProFAST</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/ProFAST/man/NCFM.html" class="external-link">NCFM</a></span>, q <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2024-09-12 20:20:48.015214 : ***** Finish CoFAST, 0.018 mins elapsed.</span></span>
<span><span class="co">#&gt; 2024-09-12 20:20:49.816489 : ***** Finish CoFAST, 0.021 mins elapsed.</span></span></code></pre></div>
<p>Then, we detect signature genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate cell-gene distance</span></span>
<span><span class="va">BC_scRNAList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_scRNAList</span>, <span class="fu">ProFAST</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/ProFAST/man/pdistance.html" class="external-link">pdistance</a></span>, reduction <span class="op">=</span> <span class="st">"ncfm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculate co-embedding distance...</span></span>
<span><span class="co">#&gt; Calculate co-embedding distance...</span></span>
<span></span>
<span><span class="co"># identify signature genes</span></span>
<span><span class="va">sg_sc_List</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_scRNAList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">CellType</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">seu</span><span class="op">$</span><span class="va">CellType</span></span>
<span>    <span class="fu"><a href="../reference/find.sig.genes.html">find.sig.genes</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       Endothelial              CAFs               PVL           B-cells </span></span>
<span><span class="co">#&gt;                41               245                46                88 </span></span>
<span><span class="co">#&gt;      Plasmablasts           T-cells           Myeloid Cancer Epithelial </span></span>
<span><span class="co">#&gt;              1453               902               206              4018 </span></span>
<span><span class="co">#&gt; Normal Epithelial </span></span>
<span><span class="co">#&gt;                24 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       Endothelial              CAFs               PVL           B-cells </span></span>
<span><span class="co">#&gt;              2778              1292              1285                99 </span></span>
<span><span class="co">#&gt;      Plasmablasts           T-cells           Myeloid Cancer Epithelial </span></span>
<span><span class="co">#&gt;                51               641               285               212 </span></span>
<span><span class="co">#&gt; Normal Epithelial </span></span>
<span><span class="co">#&gt;              1966</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">sg_sc_List</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 2</span></span>
<span><span class="co">#&gt;  $ Ref1:List of 9</span></span>
<span><span class="co">#&gt;   ..$ B-cells          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0645 0.0653 0.0657 0.0674 0.0676 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.591 0.295 0.148 0.182 0.443 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00216 0.00433 0.00173 0.01442 0.13771 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "B-cells" "B-cells" "B-cells" "B-cells" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "MS4A1" "BANK1" "SPIB" "CCR7" ...</span></span>
<span><span class="co">#&gt;   ..$ CAFs             :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0683 0.0684 0.0684 0.0684 0.0686 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.51 0.376 0.376 0.739 0.918 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00339 0.00162 0.00398 0.01107 0.01741 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "CAFs" "CAFs" "CAFs" "CAFs" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "DPT" "PDGFRA" "GJB2" "POSTN" ...</span></span>
<span><span class="co">#&gt;   ..$ Cancer Epithelial:'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0881 0.0881 0.0881 0.0881 0.0882 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.892 0.967 0.922 0.82 0.824 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0729 0.1684 0.1135 0.0483 0.0622 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Cancer Epithelial" "Cancer Epithelial" "Cancer Epithelial" "Cancer Epithelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "KRT8" "KRT7" "CLDN4" "EPCAM" ...</span></span>
<span><span class="co">#&gt;   ..$ Endothelial      :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0742 0.0743 0.0751 0.0755 0.0759 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.854 0.756 0.463 0.39 0.39 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.000573 0.00043 0.000716 0 0.000859 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Endothelial" "Endothelial" "Endothelial" "Endothelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CLEC14A" "VWF" "CLDN5" "SOX17" ...</span></span>
<span><span class="co">#&gt;   ..$ Myeloid          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0612 0.0616 0.0616 0.0616 0.0618 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.704 0.369 0.519 0.379 0.714 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.003521 0.000293 0.0022 0.00176 0.006161 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Myeloid" "Myeloid" "Myeloid" "Myeloid" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "IGSF6" "CD163" "MNDA" "ITGAX" ...</span></span>
<span><span class="co">#&gt;   ..$ Normal Epithelial:'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0823 0.0842 0.0859 0.0864 0.0865 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.417 0.292 0.25 0.417 0.75 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00143 0.00286 0.00129 0.04958 0.49707 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Normal Epithelial" "Normal Epithelial" "Normal Epithelial" "Normal Epithelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "PIGR" "KIT" "KRT14" "KRT15" ...</span></span>
<span><span class="co">#&gt;   ..$ Plasmablasts     :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0637 0.0638 0.0639 0.0641 0.0644 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.847 0.971 0.886 0.405 0.182 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.04542 0.09803 0.11239 0.00808 0.01131 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Plasmablasts" "Plasmablasts" "Plasmablasts" "Plasmablasts" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CD79A" "MZB1" "DERL3" "TNFRSF17" ...</span></span>
<span><span class="co">#&gt;   ..$ PVL              :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0743 0.0744 0.0747 0.0752 0.0752 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.8261 0.2609 0.0435 0.8043 0.7826 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.022789 0.00774 0.000717 0.057475 0.063351 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "PVL" "PVL" "PVL" "PVL" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "PDGFRB" "PTN" "TCEAL7" "ACTA2" ...</span></span>
<span><span class="co">#&gt;   ..$ T-cells          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0709 0.0711 0.0711 0.0711 0.0712 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.685 0.538 0.247 0.334 0.143 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.01797 0.01029 0.00539 0.00768 0.00327 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "T-cells" "T-cells" "T-cells" "T-cells" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CD3E" "CD3D" "CD247" "CD3G" ...</span></span>
<span><span class="co">#&gt;  $ Ref2:List of 9</span></span>
<span><span class="co">#&gt;   ..$ B-cells          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0604 0.0607 0.061 0.0638 0.0649 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.364 0.263 0.869 0.596 0.96 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.000588 0.000705 0.002938 0.006933 0.011633 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "B-cells" "B-cells" "B-cells" "B-cells" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CD19" "TCL1A" "MS4A1" "BANK1" ...</span></span>
<span><span class="co">#&gt;   ..$ CAFs             :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0699 0.07 0.0701 0.0702 0.0702 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.57 0.924 0.711 0.94 0.861 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0178 0.0692 0.0327 0.0977 0.0791 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "CAFs" "CAFs" "CAFs" "CAFs" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "DPT" "FBLN1" "IGF1" "LUM" ...</span></span>
<span><span class="co">#&gt;   ..$ Cancer Epithelial:'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0686 0.07 0.0703 0.0718 0.0721 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.0519 0.0849 0.5708 0.0896 0.7736 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00155 0.00286 0.02918 0.00512 0.04835 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Cancer Epithelial" "Cancer Epithelial" "Cancer Epithelial" "Cancer Epithelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CEACAM6" "MKI67" "SCGB2A1" "TOP2A" ...</span></span>
<span><span class="co">#&gt;   ..$ Endothelial      :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0755 0.0755 0.0755 0.0755 0.0756 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.961 0.932 0.862 0.815 0.672 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0777 0.058 0.0537 0.0566 0.0268 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Endothelial" "Endothelial" "Endothelial" "Endothelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "PECAM1" "VWF" "CLEC14A" "EGFL7" ...</span></span>
<span><span class="co">#&gt;   ..$ Myeloid          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0622 0.0623 0.0623 0.0624 0.0624 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.621 0.695 0.47 0.839 0.509 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.001442 0.005646 0.000961 0.008289 0.000961 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Myeloid" "Myeloid" "Myeloid" "Myeloid" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "IGSF6" "LYZ" "ITGAX" "AIF1" ...</span></span>
<span><span class="co">#&gt;   ..$ Normal Epithelial:'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0704 0.0704 0.0705 0.0707 0.0708 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.948 0.895 0.658 0.68 0.911 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.1439 0.1423 0.0831 0.1138 0.1891 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Normal Epithelial" "Normal Epithelial" "Normal Epithelial" "Normal Epithelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "KRT8" "TACSTD2" "SDC4" "DSP" ...</span></span>
<span><span class="co">#&gt;   ..$ Plasmablasts     :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0581 0.0694 0.0723 0.0745 0.0791 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.745 0.922 1 0.706 0.922 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00105 0.01145 0.01542 0.00631 0.01718 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Plasmablasts" "Plasmablasts" "Plasmablasts" "Plasmablasts" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "TNFRSF17" "DERL3" "MZB1" "SLAMF7" ...</span></span>
<span><span class="co">#&gt;   ..$ PVL              :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0691 0.07 0.07 0.0704 0.0708 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.2747 0.6654 0.0506 0.6833 0.6856 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.01338 0.05926 0.00614 0.11182 0.14104 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "PVL" "PVL" "PVL" "PVL" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "AVPR1A" "NDUFA4L2" "FOXC2" "MYH11" ...</span></span>
<span><span class="co">#&gt;   ..$ T-cells          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 0.0638 0.0639 0.0639 0.0642 0.0643 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.839 0.641 0.437 0.75 0.317 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00791 0.00289 0.00276 0.02246 0.00314 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "T-cells" "T-cells" "T-cells" "T-cells" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CD3E" "CD3D" "CD3G" "IL7R" ...</span></span></code></pre></div>
<p>Finally, select marker genes for each cell type from the signature
gene list.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markerList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">sg_sc_List</span>, <span class="va">marker.select</span>, overlap.max <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">markerList</span><span class="op">)</span></span>
<span><span class="co">#&gt; $Ref1</span></span>
<span><span class="co">#&gt; $Ref1$`B-cells`</span></span>
<span><span class="co">#&gt; [1] "MS4A1" "BANK1" "SPIB"  "CCR7" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref1$CAFs</span></span>
<span><span class="co">#&gt; [1] "DPT"    "PDGFRA" "GJB2"   "POSTN" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref1$`Cancer Epithelial`</span></span>
<span><span class="co">#&gt; [1] "KRT8"  "KRT7"  "CLDN4" "EPCAM"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref1$Endothelial</span></span>
<span><span class="co">#&gt; [1] "CLEC14A" "VWF"     "CLDN5"   "SOX17"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref1$Myeloid</span></span>
<span><span class="co">#&gt; [1] "IGSF6" "CD163" "MNDA"  "ITGAX"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref1$`Normal Epithelial`</span></span>
<span><span class="co">#&gt; [1] "PIGR"  "KIT"   "KRT14" "KRT15"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref1$Plasmablasts</span></span>
<span><span class="co">#&gt; [1] "CD79A"    "MZB1"     "DERL3"    "TNFRSF17"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref1$PVL</span></span>
<span><span class="co">#&gt; [1] "PDGFRB" "PTN"    "ACTA2"  "PCOLCE"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref1$`T-cells`</span></span>
<span><span class="co">#&gt; [1] "CD3E"  "CD3D"  "CD247" "CD3G" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2</span></span>
<span><span class="co">#&gt; $Ref2$`B-cells`</span></span>
<span><span class="co">#&gt; [1] "CD19"  "TCL1A" "MS4A1" "BANK1"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2$CAFs</span></span>
<span><span class="co">#&gt; [1] "DPT"   "FBLN1" "IGF1"  "LUM"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2$`Cancer Epithelial`</span></span>
<span><span class="co">#&gt; [1] "SCGB2A1" "AGR3"    "ESR1"    "FOXA1"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2$Endothelial</span></span>
<span><span class="co">#&gt; [1] "PECAM1"  "VWF"     "CLEC14A" "EGFL7"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2$Myeloid</span></span>
<span><span class="co">#&gt; [1] "IGSF6" "LYZ"   "ITGAX" "AIF1" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2$`Normal Epithelial`</span></span>
<span><span class="co">#&gt; [1] "KRT8"    "TACSTD2" "SDC4"    "DSP"    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2$Plasmablasts</span></span>
<span><span class="co">#&gt; [1] "TNFRSF17" "DERL3"    "MZB1"     "SLAMF7"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2$PVL</span></span>
<span><span class="co">#&gt; [1] "AVPR1A"   "NDUFA4L2" "MYH11"    "PDGFRB"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ref2$`T-cells`</span></span>
<span><span class="co">#&gt; [1] "CD3E" "CD3D" "CD3G" "IL7R"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="annotate-the-mob-st-data-using-caesar-and-marker-genes-from-scrna-seq-reference-data">Annotate the MOB ST data using CAESAR and marker genes from
scRNA-seq reference data<a class="anchor" aria-label="anchor" href="#annotate-the-mob-st-data-using-caesar-and-marker-genes-from-scrna-seq-reference-data"></a>
</h2>
<p>Similarly, we first calculate co-embeddings for Xenium BC dataset.
The difference is that spatial transcriptome data has spatial
coordinates and image feature information, so we can obtain image-based
spatial aware co-embeddings.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BC_XeniumList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">seu</span> <span class="op">&lt;-</span> <span class="va">BC_XeniumList</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span>    <span class="co"># the spatial coordinates</span></span>
<span>    <span class="va">pos</span> <span class="op">&lt;-</span> <span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x_centroid"</span>, <span class="st">"y_centroid"</span><span class="op">)</span><span class="op">]</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># the image feature</span></span>
<span>    <span class="va">feature_img</span> <span class="op">&lt;-</span> <span class="va">BC_feature_imgList</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span>    <span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CAESAR.coembedding.image.html">CAESAR.coembedding.image</a></span><span class="op">(</span></span>
<span>        <span class="va">seu</span>, <span class="va">feature_img</span>, <span class="va">pos</span>, reduction.name <span class="op">=</span> <span class="st">"caesar"</span>, q <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span>    <span class="va">seu</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="co">#&gt;       x_centroid y_centroid</span></span>
<span><span class="co">#&gt; 24387  2296.2025   913.7803</span></span>
<span><span class="co">#&gt; 4049    852.6599  1739.8252</span></span>
<span><span class="co">#&gt; 11570  1259.9557  1564.8037</span></span>
<span><span class="co">#&gt; 25172  2199.9682  1074.7750</span></span>
<span><span class="co">#&gt; 32617  2665.6389  1595.0457</span></span>
<span><span class="co">#&gt; 13902   923.3366  2157.2375</span></span>
<span><span class="co">#&gt; Find the adjacency matrix by bisection method...</span></span>
<span><span class="co">#&gt; Current radius is 200.5</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 0</span></span>
<span><span class="co">#&gt; Current radius is 100.75</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 66</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 17</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 4</span></span>
<span><span class="co">#&gt; Step into function</span></span>
<span><span class="co">#&gt; Centering X</span></span>
<span><span class="co">#&gt; Calculate initial values using PCA</span></span>
<span><span class="co">#&gt; Finish the initialization! </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 2, elbo= -593983.513739, delbo=-59398351373921687926824668.000000 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 3, elbo= -574548.382010, delbo=0.032720 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 4, elbo= -569948.909649, delbo=0.008005 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 5, elbo= -567200.047658, delbo=0.004823 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 6, elbo= -565256.567256, delbo=0.003426 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 7, elbo= -563815.667637, delbo=0.002549 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 8, elbo= -562743.730446, delbo=0.001901 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 9, elbo= -561912.619046, delbo=0.001477 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 10, elbo= -561232.784890, delbo=0.001210 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 11, elbo= -560646.304523, delbo=0.001045 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 12, elbo= -560118.172819, delbo=0.000942 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 13, elbo= -559625.808512, delbo=0.000879 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 14, elbo= -559155.484286, delbo=0.000840 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 15, elbo= -558699.331457, delbo=0.000816 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 16, elbo= -558254.264172, delbo=0.000797 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 17, elbo= -557820.923119, delbo=0.000776 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 18, elbo= -557402.609585, delbo=0.000750 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 19, elbo= -557003.619831, delbo=0.000716 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 20, elbo= -556627.795572, delbo=0.000675 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 21, elbo= -556277.578078, delbo=0.000629 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 22, elbo= -555953.591518, delbo=0.000582 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 23, elbo= -555654.578032, delbo=0.000538 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 24, elbo= -555377.754873, delbo=0.000498 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 25, elbo= -555119.458514, delbo=0.000465 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 26, elbo= -554875.836816, delbo=0.000439 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 27, elbo= -554643.379624, delbo=0.000419 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 28, elbo= -554419.245297, delbo=0.000404 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 29, elbo= -554201.409294, delbo=0.000393 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 30, elbo= -553988.672079, delbo=0.000384</span></span>
<span><span class="co">#&gt; 2024-09-12 20:21:11.648035 : ***** Finish calculate CAESAR embedding, 0.294 mins elapsed.</span></span>
<span><span class="co">#&gt; The spatial cooridnates are 2 dimensions</span></span>
<span><span class="co">#&gt; Find the adjacency matrix by bisection method...</span></span>
<span><span class="co">#&gt; Current radius is 200.5</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 0</span></span>
<span><span class="co">#&gt; Current radius is 100.75</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 66</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 17</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 4</span></span>
<span><span class="co">#&gt;       x_centroid y_centroid</span></span>
<span><span class="co">#&gt; 24387  1832.7835  1032.5848</span></span>
<span><span class="co">#&gt; 4049    476.8307  2294.2419</span></span>
<span><span class="co">#&gt; 11570  1655.3104   332.9922</span></span>
<span><span class="co">#&gt; 25172  1699.3427  1361.4992</span></span>
<span><span class="co">#&gt; 32617  2694.3627   286.2827</span></span>
<span><span class="co">#&gt; 13902   971.3380  1172.3937</span></span>
<span><span class="co">#&gt; Find the adjacency matrix by bisection method...</span></span>
<span><span class="co">#&gt; Current radius is 200.5</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 0</span></span>
<span><span class="co">#&gt; Current radius is 100.75</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 65</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 18</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 5</span></span>
<span><span class="co">#&gt; Step into function</span></span>
<span><span class="co">#&gt; Centering X</span></span>
<span><span class="co">#&gt; Calculate initial values using PCA</span></span>
<span><span class="co">#&gt; Finish the initialization! </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 2, elbo= -592047.437411, delbo=-59204743741083634684084602.000000 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 3, elbo= -573230.881064, delbo=0.031782 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 4, elbo= -569049.338656, delbo=0.007295 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 5, elbo= -566508.571569, delbo=0.004465 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 6, elbo= -564595.023110, delbo=0.003378 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 7, elbo= -563091.668725, delbo=0.002663 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 8, elbo= -561950.440446, delbo=0.002027 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 9, elbo= -561090.315895, delbo=0.001531 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 10, elbo= -560430.285750, delbo=0.001176 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 11, elbo= -559903.561787, delbo=0.000940 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 12, elbo= -559464.796024, delbo=0.000784 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 13, elbo= -559083.431684, delbo=0.000682 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 14, elbo= -558739.806602, delbo=0.000615 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 15, elbo= -558420.977283, delbo=0.000571 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 16, elbo= -558118.730455, delbo=0.000541 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 17, elbo= -557828.101167, delbo=0.000521 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 18, elbo= -557546.793380, delbo=0.000504 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 19, elbo= -557274.701835, delbo=0.000488 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 20, elbo= -557013.461475, delbo=0.000469 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 21, elbo= -556765.652925, delbo=0.000445 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 22, elbo= -556533.732295, delbo=0.000417 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 23, elbo= -556318.910276, delbo=0.000386 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 24, elbo= -556120.557224, delbo=0.000357 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 25, elbo= -555936.435244, delbo=0.000331 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 26, elbo= -555763.467782, delbo=0.000311 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 27, elbo= -555598.470069, delbo=0.000297 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 28, elbo= -555438.565512, delbo=0.000288 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 29, elbo= -555281.327159, delbo=0.000283 </span></span>
<span><span class="co">#&gt; Start E-step </span></span>
<span><span class="co">#&gt; Calculate XLB...Calculate M...Calculate Mu_h...Finish E-step </span></span>
<span><span class="co">#&gt; Update mu </span></span>
<span><span class="co">#&gt; Update B </span></span>
<span><span class="co">#&gt; Update Lambda </span></span>
<span><span class="co">#&gt; Update Phi </span></span>
<span><span class="co">#&gt; iter = 30, elbo= -555124.780564, delbo=0.000282</span></span>
<span><span class="co">#&gt; 2024-09-12 20:21:33.363137 : ***** Finish calculate CAESAR embedding, 0.288 mins elapsed.</span></span>
<span><span class="co">#&gt; The spatial cooridnates are 2 dimensions</span></span>
<span><span class="co">#&gt; Find the adjacency matrix by bisection method...</span></span>
<span><span class="co">#&gt; Current radius is 200.5</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 0</span></span>
<span><span class="co">#&gt; Current radius is 100.75</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 65</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 18</span></span>
<span><span class="co">#&gt; Current radius is 50.88</span></span>
<span><span class="co">#&gt; Median of neighborhoods is 5</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">BC_XeniumList</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"BC"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">BC_XeniumList</span><span class="op">)</span></span>
<span><span class="co">#&gt; $BC1</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 3000 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 302 variable features)</span></span>
<span><span class="co">#&gt;  1 dimensional reduction calculated: caesar</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $BC2</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 3000 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (302 features, 302 variable features)</span></span>
<span><span class="co">#&gt;  1 dimensional reduction calculated: caesar</span></span></code></pre></div>
<p>Subsequently, the CAESAR co-embeddings and marker gene lists from
scRNA-seq reference datasets are used to annotate the Xenium BC
data.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># convert marker list to marker frequency matrix</span></span>
<span><span class="va">marker.freq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/markerList2mat.html">markerList2mat</a></span><span class="op">(</span><span class="va">markerList</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform annotation using CAESAR and save results to Seurat object</span></span>
<span><span class="va">BC_XeniumList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>    <span class="va">BC_XeniumList</span>, <span class="va">CAESAR.annotation</span>, marker.freq <span class="op">=</span> <span class="va">marker.freq</span>,</span>
<span>    reduction.name <span class="op">=</span> <span class="st">"caesar"</span>, add.to.meta <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculate co-embedding distance...</span></span>
<span><span class="co">#&gt; Warning: package 'purrr' was built under R version 4.3.3</span></span>
<span><span class="co">#&gt; Calculate co-embedding distance...</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">BC_XeniumList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "orig.ident"             "nCount_RNA"             "nFeature_RNA"          </span></span>
<span><span class="co">#&gt;  [4] "x_centroid"             "y_centroid"             "CAESAR"                </span></span>
<span><span class="co">#&gt;  [7] "CAESARunasg"            "CAESARconf"             "dist_B.cells"          </span></span>
<span><span class="co">#&gt; [10] "dist_CAFs"              "dist_Cancer.Epithelial" "dist_Endothelial"      </span></span>
<span><span class="co">#&gt; [13] "dist_Myeloid"           "dist_Normal.Epithelial" "dist_Plasmablasts"     </span></span>
<span><span class="co">#&gt; [16] "dist_PVL"               "dist_T.cells"           "prob_B.cells"          </span></span>
<span><span class="co">#&gt; [19] "prob_CAFs"              "prob_Cancer.Epithelial" "prob_Endothelial"      </span></span>
<span><span class="co">#&gt; [22] "prob_Myeloid"           "prob_Normal.Epithelial" "prob_Plasmablasts"     </span></span>
<span><span class="co">#&gt; [25] "prob_PVL"               "prob_T.cells"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="downstream-analysis">Downstream analysis<a class="anchor" aria-label="anchor" href="#downstream-analysis"></a>
</h2>
<p>In the following, we visualize the CAESAR annotation results.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set up colors</span></span>
<span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"#fdc086"</span>, <span class="st">"#386cb0"</span>, <span class="st">"#b30000"</span>, <span class="st">"#FBEA2E"</span>, <span class="st">"#731A73"</span>,</span>
<span>        <span class="st">"#FF8C00"</span>, <span class="st">"#F898CB"</span>, <span class="st">"#4DAF4A"</span>, <span class="st">"#a6cee3"</span>, <span class="st">"#737373"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"B-cells"</span>, <span class="st">"CAFs"</span>, <span class="st">"Cancer Epithelial"</span>, <span class="st">"Endothelial"</span>, <span class="st">"Myeloid"</span>,</span>
<span>        <span class="st">"Normal Epithelial"</span>, <span class="st">"Plasmablasts"</span>, <span class="st">"PVL"</span>, <span class="st">"T-cells"</span>, <span class="st">"unassigned"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"B-cells"</span>, <span class="st">"CAFs"</span>, <span class="st">"Cancer Epithelial"</span>, <span class="st">"Endothelial"</span>, <span class="st">"Myeloid"</span>,</span>
<span>    <span class="st">"Normal Epithelial"</span>, <span class="st">"Plasmablasts"</span>, <span class="st">"PVL"</span>, <span class="st">"T-cells"</span>, <span class="st">"unassigned"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">BC_XeniumList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">CAESARunasg</span>, levels <span class="op">=</span> <span class="va">celltypes</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">pos</span> <span class="op">&lt;-</span> <span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x_centroid"</span>, <span class="st">"y_centroid"</span><span class="op">)</span><span class="op">]</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pos"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></span>
<span>    <span class="va">seu</span><span class="op">@</span><span class="va">reductions</span><span class="op">[[</span><span class="st">"pos"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateDimReducObject.html" class="external-link">CreateDimReducObject</a></span><span class="op">(</span></span>
<span>        embeddings <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span>,</span>
<span>        key <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pos"</span>, <span class="st">"_"</span><span class="op">)</span>, assay <span class="op">=</span> <span class="st">"RNA"</span></span>
<span>    <span class="op">)</span></span>
<span>    <span class="va">seu</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>First, we visualize the CAESAR annotation account for
‘unassigned’.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seu</span>, reduction <span class="op">=</span> <span class="st">"pos"</span>, cols <span class="op">=</span> <span class="va">cols</span>, pt.size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">plots</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="XeniumBC_files/figure-html/unnamed-chunk-12-1.png" width="1152"></p>
<p>The confidence level of the CAESAR annotation can be visualized
by</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span></span>
<span>        <span class="va">seu</span>,</span>
<span>        reduction <span class="op">=</span> <span class="st">"pos"</span>, features <span class="op">=</span> <span class="st">"CAESARconf"</span>, pt.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"lightgrey"</span><span class="op">)</span>, min.cutoff <span class="op">=</span> <span class="fl">0.0</span>, max.cutoff <span class="op">=</span> <span class="fl">1.0</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">plots</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="XeniumBC_files/figure-html/unnamed-chunk-13-1.png" width="1152"></p>
<p>Next, we detect and visualize the signature genes for each cell
type.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sg_List</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="va">find.sig.genes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">sg_List</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 2</span></span>
<span><span class="co">#&gt;  $ BC1:List of 9</span></span>
<span><span class="co">#&gt;   ..$ CAFs             :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 22.1 22.3 22.4 22.4 22.5 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.643 0.632 0.353 0.711 0.327 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0794 0.0951 0.056 0.1397 0.0435 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "CAFs" "CAFs" "CAFs" "CAFs" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "FBLN1" "PTGDS" "DPT" "CCDC80" ...</span></span>
<span><span class="co">#&gt;   ..$ Cancer Epithelial:'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 16.1 16.1 16.2 16.2 16.2 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.363 0.926 0.902 0.814 0.965 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0829 0.2797 0.2698 0.2174 0.3746 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Cancer Epithelial" "Cancer Epithelial" "Cancer Epithelial" "Cancer Epithelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "LYPD3" "AGR3" "MLPH" "SCD" ...</span></span>
<span><span class="co">#&gt;   ..$ Endothelial      :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 22.8 22.9 22.9 23 23 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.465 0.774 0.292 0.341 0.27 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0169 0.0451 0.0105 0.018 0.0177 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Endothelial" "Endothelial" "Endothelial" "Endothelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "HOXD9" "VWF" "EGFL7" "IL3RA" ...</span></span>
<span><span class="co">#&gt;   ..$ Myeloid          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 27.5 27.6 27.6 28.1 28.1 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.648 0.617 0.606 0.425 0.896 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0242 0.0492 0.0474 0.0392 0.1332 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Myeloid" "Myeloid" "Myeloid" "Myeloid" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "ITGAX" "HAVCR2" "IGSF6" "CD86" ...</span></span>
<span><span class="co">#&gt;   ..$ Normal Epithelial:'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 21.3 22.3 23 23.4 23.5 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.452 0.637 0.459 0.325 0.382 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0383 0.1164 0.0552 0.0616 0.1115 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Normal Epithelial" "Normal Epithelial" "Normal Epithelial" "Normal Epithelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "PIGR" "KRT15" "KIT" "KRT6B" ...</span></span>
<span><span class="co">#&gt;   ..$ Plasmablasts     :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 29.1 29.7 30.7 31.3 32.3 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.667 0.444 0.556 0.852 0.556 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00303 0.00774 0.00774 0.0111 0.01043 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Plasmablasts" "Plasmablasts" "Plasmablasts" "Plasmablasts" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CD79A" "DERL3" "TNFRSF17" "MZB1" ...</span></span>
<span><span class="co">#&gt;   ..$ PVL              :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 23.2 23.3 23.4 23.5 23.6 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.887 0.604 0.453 0.491 0.585 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.195 0.185 0.116 0.166 0.223 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "PVL" "PVL" "PVL" "PVL" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "PDGFRB" "ZEB1" "SFRP4" "EGFR" ...</span></span>
<span><span class="co">#&gt;   ..$ T-cells          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 29.7 29.7 29.9 30.7 30.8 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.616 0.479 0.904 0.384 0.425 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0123 0.00922 0.03075 0.01435 0.01127 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "T-cells" "T-cells" "T-cells" "T-cells" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CD3G" "CD3D" "CD3E" "CD69" ...</span></span>
<span><span class="co">#&gt;   ..$ unassigned       :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 25.3 25.3 25.4 25.4 25.4 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.691 0.59 0.511 0.429 0.784 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.349 0.244 0.138 0.132 0.573 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "unassigned" "unassigned" "unassigned" "unassigned" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "MYLK" "KRT14" "KRT5" "MYH11" ...</span></span>
<span><span class="co">#&gt;  $ BC2:List of 10</span></span>
<span><span class="co">#&gt;   ..$ B-cells          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 35.9 37.8 46 47 48.9 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.667 0.667 0.667 1 0.667 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00334 0.00634 0.00968 0.01668 0.01568 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "B-cells" "B-cells" "B-cells" "B-cells" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "TCL1A" "LILRA4" "SPIB" "PLD4" ...</span></span>
<span><span class="co">#&gt;   ..$ CAFs             :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 21.4 21.7 21.7 21.7 21.8 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.72 0.741 0.478 0.763 0.237 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0755 0.1311 0.0679 0.1293 0.0322 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "CAFs" "CAFs" "CAFs" "CAFs" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "FBLN1" "PDGFRA" "CRISPLD2" "CCDC80" ...</span></span>
<span><span class="co">#&gt;   ..$ Cancer Epithelial:'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 16.7 16.7 16.7 16.7 16.7 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.915 0.892 0.973 0.384 0.758 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.2341 0.2523 0.3097 0.0785 0.1941 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Cancer Epithelial" "Cancer Epithelial" "Cancer Epithelial" "Cancer Epithelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "AGR3" "MLPH" "ESR1" "LYPD3" ...</span></span>
<span><span class="co">#&gt;   ..$ Endothelial      :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 22.9 23 23 23 23 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.481 0.276 0.338 0.59 0.429 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.01362 0.00932 0.01541 0.03907 0.01828 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Endothelial" "Endothelial" "Endothelial" "Endothelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "HOXD9" "EGFL7" "SOX17" "KDR" ...</span></span>
<span><span class="co">#&gt;   ..$ Myeloid          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 27.5 27.6 27.6 27.7 27.8 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.592 0.512 0.458 0.408 0.905 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.0204 0.0447 0.0414 0.0297 0.1333 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Myeloid" "Myeloid" "Myeloid" "Myeloid" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "ITGAX" "HAVCR2" "IGSF6" "CD86" ...</span></span>
<span><span class="co">#&gt;   ..$ Normal Epithelial:'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 20.5 20.9 21 21 21.4 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.601 0.255 0.415 0.707 0.867 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.1031 0.0484 0.0861 0.2361 0.3855 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Normal Epithelial" "Normal Epithelial" "Normal Epithelial" "Normal Epithelial" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "KRT15" "PIGR" "KRT16" "KRT14" ...</span></span>
<span><span class="co">#&gt;   ..$ Plasmablasts     :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 27.4 27.8 28.3 29.4 31.9 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.56 0.52 0.8 0.44 0.8 0.4 0.88 0.48 0.08 0.36 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.00403 0.00403 0.00975 0.00605 0.03361 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "Plasmablasts" "Plasmablasts" "Plasmablasts" "Plasmablasts" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CD79A" "TNFRSF17" "MZB1" "DERL3" ...</span></span>
<span><span class="co">#&gt;   ..$ PVL              :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 21.9 22.1 22.3 22.3 22.4 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.7255 0.4706 0.4314 0.1765 0.0588 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.2197 0.1109 0.1838 0.0231 0.0309 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "PVL" "PVL" "PVL" "PVL" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CAV1" "SFRP4" "ZEB1" "AVPR1A" ...</span></span>
<span><span class="co">#&gt;   ..$ T-cells          :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 30.7 31 31 31.6 31.8 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.84 0.531 0.691 0.531 0.333 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.024 0.0164 0.0319 0.0116 0.0134 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "T-cells" "T-cells" "T-cells" "T-cells" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "CD3E" "CD3G" "IL7R" "CD3D" ...</span></span>
<span><span class="co">#&gt;   ..$ unassigned       :'data.frame':    302 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ distance        : num [1:302] 24.9 24.9 25 25.1 25.1 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop       : num [1:302] 0.64 0.378 0.724 0.429 0.52 ...</span></span>
<span><span class="co">#&gt;   .. ..$ expr.prop.others: num [1:302] 0.364 0.147 0.558 0.147 0.383 ...</span></span>
<span><span class="co">#&gt;   .. ..$ label           : chr [1:302] "unassigned" "unassigned" "unassigned" "unassigned" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene            : chr [1:302] "MYLK" "MYH11" "ACTA2" "KRT5" ...</span></span></code></pre></div>
<p>We remove unwanted variation by</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dist_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"dist_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"-|/| "</span>, <span class="st">"."</span>, <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">celltypes</span>, <span class="st">"unassigned"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">distList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span>, <span class="va">dist_names</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seuInt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CAESAR.RUV.html">CAESAR.RUV</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="va">distList</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, species <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metaInt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span>, <span class="st">"CAESARunasg"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">metaInt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"CAESARunasg"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">metaInt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seuInt</span><span class="op">)</span></span>
<span><span class="va">seuInt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AddMetaData.html" class="external-link">AddMetaData</a></span><span class="op">(</span><span class="va">seuInt</span>, <span class="va">metaInt</span>, col.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">metaInt</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seuInt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">seuInt</span><span class="op">$</span><span class="va">CAESARunasg</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cols</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">seuInt</span><span class="op">)</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 302 features across 6000 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: CAESAR (302 features, 0 variable features)</span></span></code></pre></div>
<p>Then, we can visualize the top three signature genes by a dot
plot.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># obtain the top three signature genes</span></span>
<span><span class="va">celltypes_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">celltypes</span>, <span class="st">"unassigned"</span><span class="op">)</span></span>
<span><span class="va">top3sgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Intsg.html">Intsg</a></span><span class="op">(</span><span class="va">sg_List</span>, <span class="fl">3</span><span class="op">)</span><span class="op">[</span><span class="va">celltypes_plot</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">top3sgs</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`B-cells`</span></span>
<span><span class="co">#&gt; [1] "TCL1A"  "LILRA4" "SPIB"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $CAFs</span></span>
<span><span class="co">#&gt; [1] "FBLN1"  "PTGDS"  "CCDC80"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`Cancer Epithelial`</span></span>
<span><span class="co">#&gt; [1] "AGR3"  "LYPD3" "MLPH" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Endothelial</span></span>
<span><span class="co">#&gt; [1] "HOXD9" "EGFL7" "VWF"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Myeloid</span></span>
<span><span class="co">#&gt; [1] "ITGAX"  "HAVCR2" "IGSF6" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`Normal Epithelial`</span></span>
<span><span class="co">#&gt; [1] "PIGR"  "KRT15" "KRT6B"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Plasmablasts</span></span>
<span><span class="co">#&gt; [1] "CD79A"    "TNFRSF17" "DERL3"   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $PVL</span></span>
<span><span class="co">#&gt; [1] "ZEB1"  "SFRP4" "CAV1" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`T-cells`</span></span>
<span><span class="co">#&gt; [1] "CD3G" "CD3E" "CD3D"</span></span>
<span></span>
<span><span class="va">sg_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">top3sgs</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span></span>
<span>    <span class="va">seuInt</span>,</span>
<span>    idents <span class="op">=</span> <span class="va">celltypes_plot</span>, col.min <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, col.max <span class="op">=</span> <span class="fl">2</span>, dot.scale <span class="op">=</span> <span class="fl">7</span>,</span>
<span>    features <span class="op">=</span> <span class="va">sg_features</span>, scale.min <span class="op">=</span> <span class="fl">0</span>, scale.max <span class="op">=</span> <span class="fl">30</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span>, angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="XeniumBC_files/figure-html/unnamed-chunk-16-1.png" width="1152"></p>
<p>Next, we calculate the UMAP projections of co-embeddings of cells and
the selected signature genes.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate coumap</span></span>
<span><span class="va">BC_XeniumList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>    <span class="va">BC_XeniumList</span>, <span class="va">CoUMAP</span>, reduction <span class="op">=</span> <span class="st">"caesar"</span>,</span>
<span>    reduction.name <span class="op">=</span> <span class="st">"caesarUMAP"</span>, gene.set <span class="op">=</span> <span class="va">sg_features</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">df_gene_label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">top3sgs</span><span class="op">)</span>,</span>
<span>    label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">top3sgs</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BC_XeniumList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/CoUMAP.plot.html">CoUMAP.plot</a></span><span class="op">(</span></span>
<span>        <span class="va">seu</span>, reduction <span class="op">=</span> <span class="st">"caesarUMAP"</span>, gene_txtdata <span class="op">=</span> <span class="va">df_gene_label</span>,</span>
<span>        cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span> <span class="op">=</span> <span class="st">"#000000"</span>, <span class="va">cols</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">plots</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="XeniumBC_files/figure-html/unnamed-chunk-17-1.png" width="1152"></p>
</div>
<div class="section level2">
<h2 id="enrichment-analysis">Enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h2>
<p>Next, we show how to use CAESAR for enrichment analysis. Here we
choose GOBP pathways as an example. Let’s first get some pathways.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathway_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span>, category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"GO:BP"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">gs_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">gene_symbol</span>, <span class="va">common_genes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/enframe.html" class="external-link">deframe</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">n.pathway_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">pathway_list</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="va">pathway_list</span> <span class="op">&lt;-</span> <span class="va">pathway_list</span><span class="op">[</span><span class="va">n.pathway_list</span> <span class="op">&gt;=</span> <span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pathway_list</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $GOBP_ACTIN_CYTOSKELETON_REORGANIZATION</span></span>
<span><span class="co">#&gt; [1] "CSF3"    "CTTN"    "DAPK3"   "KIT"     "PDGFRA"  "RAPGEF3"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $GOBP_ACTIN_FILAMENT_BASED_MOVEMENT</span></span>
<span><span class="co">#&gt; [1] "ACTA2" "CAV1"  "DSC2"  "DSP"   "JUP"   "MYO5B" "STC1" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $GOBP_ACTIN_FILAMENT_BASED_PROCESS</span></span>
<span><span class="co">#&gt;  [1] "ACTA2"    "AIF1"     "AQP1"     "CAV1"     "CCR7"     "CDC42EP1"</span></span>
<span><span class="co">#&gt;  [7] "CSF3"     "CTTN"     "CXCL12"   "DAPK3"    "DSC2"     "DSP"     </span></span>
<span><span class="co">#&gt; [13] "EDN1"     "ENAH"     "FLNB"     "JUP"      "KIT"      "MYH11"   </span></span>
<span><span class="co">#&gt; [19] "MYO5B"    "PDGFRA"   "PDGFRB"   "RAPGEF3"  "RHOH"     "SDC4"    </span></span>
<span><span class="co">#&gt; [25] "SFRP1"    "STC1"     "SVIL"     "TACSTD2"  "TYROBP"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $GOBP_ACTIN_FILAMENT_BUNDLE_ORGANIZATION</span></span>
<span><span class="co">#&gt; [1] "AIF1"    "RAPGEF3" "SDC4"    "SFRP1"   "TACSTD2"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $GOBP_ACTIN_FILAMENT_ORGANIZATION</span></span>
<span><span class="co">#&gt;  [1] "AIF1"     "CCR7"     "CDC42EP1" "CSF3"     "CTTN"     "CXCL12"  </span></span>
<span><span class="co">#&gt;  [7] "ENAH"     "MYO5B"    "RAPGEF3"  "RHOH"     "SDC4"     "SFRP1"   </span></span>
<span><span class="co">#&gt; [13] "SVIL"     "TACSTD2" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $GOBP_ACTIN_FILAMENT_POLYMERIZATION</span></span>
<span><span class="co">#&gt; [1] "AIF1"     "CCR7"     "CDC42EP1" "CSF3"     "CTTN"     "SVIL"</span></span></code></pre></div>
<p>Then, we can test whether those pathways are enriched in BC1
section.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seuBC1</span> <span class="op">&lt;-</span> <span class="va">BC_XeniumList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">df_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CAESAR.enrich.pathway.html">CAESAR.enrich.pathway</a></span><span class="op">(</span></span>
<span>    <span class="va">seuBC1</span>, pathway.list <span class="op">=</span> <span class="va">pathway_list</span>, reduction <span class="op">=</span> <span class="st">"caesar"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Only the approximate p-values based on asymptotic theory are calculated as perm.num is set as 0.</span></span>
<span></span>
<span><span class="co"># obtain significant enriched pathways</span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="va">pathway_list</span><span class="op">[</span><span class="va">df_enrich</span><span class="op">$</span><span class="va">asy.wei.pval.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span></span></code></pre></div>
<p>Next, we can calculate the spot level enrichment scores and detect
differentially enriched pathways.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enrich.score.BC1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CAESAR.enrich.score.html">CAESAR.enrich.score</a></span><span class="op">(</span><span class="va">seuBC1</span>, <span class="va">pathways</span><span class="op">)</span></span>
<span><span class="co">#&gt; There are 223 pathways. The largest pathway has 98 genes.</span></span>
<span><span class="co">#&gt; Pathways with 5 genes finished, which includes 32 pathways, elapsed time is 0.89s.</span></span>
<span><span class="co">#&gt; Pathways with 6 genes finished, which includes 26 pathways, elapsed time is 0.72s.</span></span>
<span><span class="co">#&gt; Pathways with 7 genes finished, which includes 20 pathways, elapsed time is 0.59s.</span></span>
<span><span class="co">#&gt; Pathways with 8 genes finished, which includes 15 pathways, elapsed time is 0.36s.</span></span>
<span><span class="co">#&gt; Pathways with 9 genes finished, which includes 7 pathways, elapsed time is 0.8s.</span></span>
<span><span class="co">#&gt; Pathways with 10 genes finished, which includes 7 pathways, elapsed time is 0.19s.</span></span>
<span><span class="co">#&gt; Pathways with 11 genes finished, which includes 9 pathways, elapsed time is 0.26s.</span></span>
<span><span class="co">#&gt; Pathways with 12 genes finished, which includes 6 pathways, elapsed time is 0.17s.</span></span>
<span><span class="co">#&gt; Pathways with 13 genes finished, which includes 5 pathways, elapsed time is 0.16s.</span></span>
<span><span class="co">#&gt; Pathways with 14 genes finished, which includes 6 pathways, elapsed time is 0.14s.</span></span>
<span><span class="co">#&gt; Pathways with 15 genes finished, which includes 3 pathways, elapsed time is 0.14s.</span></span>
<span><span class="co">#&gt; Pathways with 16 genes finished, which includes 2 pathways, elapsed time is 0.08s.</span></span>
<span><span class="co">#&gt; Pathways with 17 genes finished, which includes 2 pathways, elapsed time is 0.11s.</span></span>
<span><span class="co">#&gt; Pathways with 18 genes finished, which includes 3 pathways, elapsed time is 0.11s.</span></span>
<span><span class="co">#&gt; Pathways with 19 genes finished, which includes 5 pathways, elapsed time is 0.15s.</span></span>
<span><span class="co">#&gt; Pathways with 20 genes finished, which includes 4 pathways, elapsed time is 0.18s.</span></span>
<span><span class="co">#&gt; Pathways with 21 genes finished, which includes 4 pathways, elapsed time is 0.12s.</span></span>
<span><span class="co">#&gt; Pathways with 22 genes finished, which includes 4 pathways, elapsed time is 0.16s.</span></span>
<span><span class="co">#&gt; Pathways with 23 genes finished, which includes 1 pathways, elapsed time is 0.04s.</span></span>
<span><span class="co">#&gt; Pathways with 24 genes finished, which includes 4 pathways, elapsed time is 0.14s.</span></span>
<span><span class="co">#&gt; Pathways with 25 genes finished, which includes 5 pathways, elapsed time is 0.16s.</span></span>
<span><span class="co">#&gt; Pathways with 26 genes finished, which includes 1 pathways, elapsed time is 0.08s.</span></span>
<span><span class="co">#&gt; Pathways with 27 genes finished, which includes 1 pathways, elapsed time is 0.05s.</span></span>
<span><span class="co">#&gt; Pathways with 29 genes finished, which includes 1 pathways, elapsed time is 0.08s.</span></span>
<span><span class="co">#&gt; Pathways with 30 genes finished, which includes 5 pathways, elapsed time is 0.15s.</span></span>
<span><span class="co">#&gt; Pathways with 31 genes finished, which includes 1 pathways, elapsed time is 0.05s.</span></span>
<span><span class="co">#&gt; Pathways with 32 genes finished, which includes 4 pathways, elapsed time is 0.16s.</span></span>
<span><span class="co">#&gt; Pathways with 34 genes finished, which includes 1 pathways, elapsed time is 0.04s.</span></span>
<span><span class="co">#&gt; Pathways with 35 genes finished, which includes 1 pathways, elapsed time is 0.05s.</span></span>
<span><span class="co">#&gt; Pathways with 36 genes finished, which includes 3 pathways, elapsed time is 0.14s.</span></span>
<span><span class="co">#&gt; Pathways with 37 genes finished, which includes 4 pathways, elapsed time is 0.16s.</span></span>
<span><span class="co">#&gt; Pathways with 38 genes finished, which includes 1 pathways, elapsed time is 0.03s.</span></span>
<span><span class="co">#&gt; Pathways with 39 genes finished, which includes 2 pathways, elapsed time is 0.11s.</span></span>
<span><span class="co">#&gt; Pathways with 41 genes finished, which includes 3 pathways, elapsed time is 0.12s.</span></span>
<span><span class="co">#&gt; Pathways with 42 genes finished, which includes 1 pathways, elapsed time is 0.07s.</span></span>
<span><span class="co">#&gt; Pathways with 43 genes finished, which includes 1 pathways, elapsed time is 0.04s.</span></span>
<span><span class="co">#&gt; Pathways with 44 genes finished, which includes 2 pathways, elapsed time is 0.1s.</span></span>
<span><span class="co">#&gt; Pathways with 47 genes finished, which includes 1 pathways, elapsed time is 0.08s.</span></span>
<span><span class="co">#&gt; Pathways with 48 genes finished, which includes 1 pathways, elapsed time is 0.06s.</span></span>
<span><span class="co">#&gt; Pathways with 49 genes finished, which includes 1 pathways, elapsed time is 0.05s.</span></span>
<span><span class="co">#&gt; Pathways with 50 genes finished, which includes 1 pathways, elapsed time is 0.08s.</span></span>
<span><span class="co">#&gt; Pathways with 52 genes finished, which includes 3 pathways, elapsed time is 0.14s.</span></span>
<span><span class="co">#&gt; Pathways with 55 genes finished, which includes 1 pathways, elapsed time is 0.05s.</span></span>
<span><span class="co">#&gt; Pathways with 56 genes finished, which includes 1 pathways, elapsed time is 0.07s.</span></span>
<span><span class="co">#&gt; Pathways with 57 genes finished, which includes 1 pathways, elapsed time is 0.06s.</span></span>
<span><span class="co">#&gt; Pathways with 60 genes finished, which includes 3 pathways, elapsed time is 0.12s.</span></span>
<span><span class="co">#&gt; Pathways with 63 genes finished, which includes 1 pathways, elapsed time is 0.04s.</span></span>
<span><span class="co">#&gt; Pathways with 73 genes finished, which includes 1 pathways, elapsed time is 0.03s.</span></span>
<span><span class="co">#&gt; Pathways with 78 genes finished, which includes 1 pathways, elapsed time is 0.05s.</span></span>
<span><span class="co">#&gt; Pathways with 79 genes finished, which includes 1 pathways, elapsed time is 0.06s.</span></span>
<span><span class="co">#&gt; Pathways with 86 genes finished, which includes 1 pathways, elapsed time is 0.06s.</span></span>
<span><span class="co">#&gt; Pathways with 87 genes finished, which includes 2 pathways, elapsed time is 0.09s.</span></span>
<span><span class="co">#&gt; Pathways with 98 genes finished, which includes 1 pathways, elapsed time is 0.07s.</span></span>
<span></span>
<span><span class="va">dep.pvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CAESAR.CTDEP.html">CAESAR.CTDEP</a></span><span class="op">(</span><span class="va">seuBC1</span>, <span class="va">enrich.score.BC1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dep.pvals</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                                                                                                                         CAFs</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                            2.995968e-138</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                              1.002358e-87</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                  7.012863e-118</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS  3.263156e-77</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                     1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                 5.080902e-141</span></span>
<span><span class="co">#&gt;                                                                                                                                Cancer Epithelial</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                                 1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                                  1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                       1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS      1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                         9.934717e-50</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                      1.000000e+00</span></span>
<span><span class="co">#&gt;                                                                                                                                 Endothelial</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                            9.062098e-26</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                             6.990295e-42</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                  1.329502e-07</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS 7.945749e-23</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                    1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                 1.210054e-69</span></span>
<span><span class="co">#&gt;                                                                                                                                      Myeloid</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                             2.112662e-37</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                              6.539622e-96</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                  2.594184e-158</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS 1.799025e-105</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                     1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                  9.999990e-01</span></span>
<span><span class="co">#&gt;                                                                                                                                Normal Epithelial</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                                 1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                                  2.835952e-02</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                       1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS      1.248807e-06</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                         1.014166e-11</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                      9.999960e-01</span></span>
<span><span class="co">#&gt;                                                                                                                                Plasmablasts</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                            5.299316e-10</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                             1.273776e-18</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                  9.008437e-29</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS 4.040162e-15</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                    1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                 9.975925e-01</span></span>
<span><span class="co">#&gt;                                                                                                                                         PVL</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                            5.852153e-14</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                             7.499290e-11</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                  6.030867e-06</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS 4.442089e-10</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                    9.989642e-01</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                 1.775810e-17</span></span>
<span><span class="co">#&gt;                                                                                                                                     T-cells</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                            2.299836e-36</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                             3.283280e-45</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                  5.660293e-68</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS 3.055402e-41</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                    1.000000e+00</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                 1.986444e-07</span></span>
<span><span class="co">#&gt;                                                                                                                                  unassigned</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATED_T_CELL_PROLIFERATION                                                                                            4.296084e-05</span></span>
<span><span class="co">#&gt; GOBP_ACTIVATION_OF_IMMUNE_RESPONSE                                                                                             3.944759e-12</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE                                                                                                  6.614251e-01</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_IMMUNE_RESPONSE_BASED_ON_SOMATIC_RECOMBINATION_OF_IMMUNE_RECEPTORS_BUILT_FROM_IMMUNOGLOBULIN_SUPERFAMILY_DOMAINS 2.776858e-29</span></span>
<span><span class="co">#&gt; GOBP_ADAPTIVE_THERMOGENESIS                                                                                                    4.951793e-22</span></span>
<span><span class="co">#&gt; GOBP_ALCOHOL_METABOLIC_PROCESS                                                                                                 9.999998e-01</span></span></code></pre></div>
<p>We can visualize the spatial heatmap of enrichment score.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seuBC1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AddMetaData.html" class="external-link">AddMetaData</a></span><span class="op">(</span><span class="va">seuBC1</span>, <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">enrich.score.BC1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pathway</span> <span class="op">&lt;-</span> <span class="st">"GOBP_VASCULATURE_DEVELOPMENT"</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">seuBC1</span>, features <span class="op">=</span> <span class="va">pathway</span>, reduction <span class="op">=</span> <span class="st">"pos"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span></span>
<span>        colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#fff7f3"</span>, <span class="st">"#fcc5c0"</span>, <span class="st">"#f768a1"</span>, <span class="st">"#ae017e"</span>, <span class="st">"#49006a"</span><span class="op">)</span>,</span>
<span>        values <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/rescale.html" class="external-link">rescale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0.25</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        legend.position <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>        legend.justification <span class="op">=</span> <span class="st">"center"</span>,</span>
<span>        legend.box <span class="op">=</span> <span class="st">"vertical"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="XeniumBC_files/figure-html/unnamed-chunk-21-1.png" width="1152"></p>
<details><summary><strong>Session Info</strong>
</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 22631)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Chinese (Simplified)_China.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Chinese (Simplified)_China.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Chinese (Simplified)_China.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                               </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Chinese (Simplified)_China.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Shanghai</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] purrr_1.0.2        dplyr_1.1.4        msigdbr_7.5.1      ggplot2_3.5.1     </span></span>
<span><span class="co">#&gt; [5] ProFAST_1.4        gtools_3.9.5       SeuratObject_4.1.4 Seurat_4.4.0      </span></span>
<span><span class="co">#&gt; [9] CAESAR.Suite_0.1.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] fs_1.6.4                    matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt;   [3] spatstat.sparse_3.1-0       bitops_1.0-8               </span></span>
<span><span class="co">#&gt;   [5] httr_1.4.7                  RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;   [7] tools_4.3.1                 sctransform_0.4.1          </span></span>
<span><span class="co">#&gt;   [9] backports_1.5.0             utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [11] R6_2.5.1                    lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [13] uwot_0.2.2                  withr_3.0.1                </span></span>
<span><span class="co">#&gt;  [15] sp_2.1-4                    prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt;  [17] gridExtra_2.3               progressr_0.14.0           </span></span>
<span><span class="co">#&gt;  [19] cli_3.6.3                   Biobase_2.62.0             </span></span>
<span><span class="co">#&gt;  [21] textshaping_0.4.0           spatstat.explore_3.3-2     </span></span>
<span><span class="co">#&gt;  [23] labeling_0.4.3              sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [25] mvtnorm_1.2-6               spatstat.data_3.1-2        </span></span>
<span><span class="co">#&gt;  [27] proxy_0.4-27                ggridges_0.5.6             </span></span>
<span><span class="co">#&gt;  [29] pbapply_1.7-2               pkgdown_2.0.6              </span></span>
<span><span class="co">#&gt;  [31] systemfonts_1.1.0           harmony_1.2.1              </span></span>
<span><span class="co">#&gt;  [33] scater_1.30.1               parallelly_1.38.0          </span></span>
<span><span class="co">#&gt;  [35] readxl_1.4.3                rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;  [37] RSQLite_2.3.7               FNN_1.1.4                  </span></span>
<span><span class="co">#&gt;  [39] generics_0.1.3              ica_1.0-3                  </span></span>
<span><span class="co">#&gt;  [41] spatstat.random_3.3-1       car_3.1-2                  </span></span>
<span><span class="co">#&gt;  [43] Matrix_1.6-1.1              ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt;  [45] fansi_1.0.6                 DescTools_0.99.56          </span></span>
<span><span class="co">#&gt;  [47] S4Vectors_0.40.2            abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [49] lifecycle_1.0.4             yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [51] CompQuadForm_1.4.3          carData_3.0-5              </span></span>
<span><span class="co">#&gt;  [53] SummarizedExperiment_1.32.0 SparseArray_1.2.4          </span></span>
<span><span class="co">#&gt;  [55] BiocFileCache_2.10.2        Rtsne_0.17                 </span></span>
<span><span class="co">#&gt;  [57] grid_4.3.1                  blob_1.2.4                 </span></span>
<span><span class="co">#&gt;  [59] promises_1.3.0              crayon_1.5.3               </span></span>
<span><span class="co">#&gt;  [61] GiRaF_1.0.1                 miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [63] lattice_0.21-8              beachmat_2.18.1            </span></span>
<span><span class="co">#&gt;  [65] cowplot_1.1.3               KEGGREST_1.42.0            </span></span>
<span><span class="co">#&gt;  [67] pillar_1.9.0                knitr_1.48                 </span></span>
<span><span class="co">#&gt;  [69] GenomicRanges_1.54.1        boot_1.3-28.1              </span></span>
<span><span class="co">#&gt;  [71] gld_2.6.6                   future.apply_1.11.2        </span></span>
<span><span class="co">#&gt;  [73] codetools_0.2-19            leiden_0.4.3.1             </span></span>
<span><span class="co">#&gt;  [75] glue_1.7.0                  spatstat.univar_3.0-0      </span></span>
<span><span class="co">#&gt;  [77] data.table_1.16.0           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [79] png_0.1-8                   spam_2.10-0                </span></span>
<span><span class="co">#&gt;  [81] org.Mm.eg.db_3.18.0         cellranger_1.1.0           </span></span>
<span><span class="co">#&gt;  [83] gtable_0.3.5                cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [85] xfun_0.47                   S4Arrays_1.2.1             </span></span>
<span><span class="co">#&gt;  [87] mime_0.12                   survival_3.5-5             </span></span>
<span><span class="co">#&gt;  [89] SingleCellExperiment_1.24.0 fitdistrplus_1.2-1         </span></span>
<span><span class="co">#&gt;  [91] ROCR_1.0-11                 nlme_3.1-162               </span></span>
<span><span class="co">#&gt;  [93] bit64_4.0.5                 filelock_1.0.3             </span></span>
<span><span class="co">#&gt;  [95] progress_1.2.3              RcppAnnoy_0.0.22           </span></span>
<span><span class="co">#&gt;  [97] GenomeInfoDb_1.38.8         bslib_0.8.0                </span></span>
<span><span class="co">#&gt;  [99] irlba_2.3.5.1               vipor_0.4.7                </span></span>
<span><span class="co">#&gt; [101] KernSmooth_2.23-21          colorspace_2.1-1           </span></span>
<span><span class="co">#&gt; [103] BiocGenerics_0.48.1         DBI_1.2.3                  </span></span>
<span><span class="co">#&gt; [105] ade4_1.7-22                 Exact_3.3                  </span></span>
<span><span class="co">#&gt; [107] tidyselect_1.2.1            DR.SC_3.4                  </span></span>
<span><span class="co">#&gt; [109] curl_5.2.2                  bit_4.0.5                  </span></span>
<span><span class="co">#&gt; [111] compiler_4.3.1              BiocNeighbors_1.20.2       </span></span>
<span><span class="co">#&gt; [113] expm_1.0-0                  xml2_1.3.6                 </span></span>
<span><span class="co">#&gt; [115] desc_1.4.3                  DelayedArray_0.28.0        </span></span>
<span><span class="co">#&gt; [117] plotly_4.10.4               scales_1.3.0               </span></span>
<span><span class="co">#&gt; [119] lmtest_0.9-40               rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt; [121] stringr_1.5.1               digest_0.6.37              </span></span>
<span><span class="co">#&gt; [123] goftest_1.2-3               spatstat.utils_3.1-0       </span></span>
<span><span class="co">#&gt; [125] rmarkdown_2.28              XVector_0.42.0             </span></span>
<span><span class="co">#&gt; [127] htmltools_0.5.8.1           pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [129] sparseMatrixStats_1.14.0    MatrixGenerics_1.14.0      </span></span>
<span><span class="co">#&gt; [131] highr_0.11                  dbplyr_2.5.0               </span></span>
<span><span class="co">#&gt; [133] fastmap_1.2.0               rlang_1.1.4                </span></span>
<span><span class="co">#&gt; [135] htmlwidgets_1.6.4           ggthemes_5.1.0             </span></span>
<span><span class="co">#&gt; [137] shiny_1.9.1                 DelayedMatrixStats_1.24.0  </span></span>
<span><span class="co">#&gt; [139] farver_2.1.2                jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [141] zoo_1.8-12                  jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt; [143] BiocParallel_1.36.0         mclust_6.1.1               </span></span>
<span><span class="co">#&gt; [145] BiocSingular_1.18.0         RCurl_1.98-1.16            </span></span>
<span><span class="co">#&gt; [147] magrittr_2.0.3              scuttle_1.12.0             </span></span>
<span><span class="co">#&gt; [149] GenomeInfoDbData_1.2.11     dotCall64_1.1-1            </span></span>
<span><span class="co">#&gt; [151] patchwork_1.2.0             munsell_0.5.1              </span></span>
<span><span class="co">#&gt; [153] Rcpp_1.0.13                 babelgene_22.9             </span></span>
<span><span class="co">#&gt; [155] viridis_0.6.5               reticulate_1.38.0          </span></span>
<span><span class="co">#&gt; [157] furrr_0.3.1                 stringi_1.8.4              </span></span>
<span><span class="co">#&gt; [159] rootSolve_1.8.2.4           zlibbioc_1.48.2            </span></span>
<span><span class="co">#&gt; [161] MASS_7.3-60                 org.Hs.eg.db_3.18.0        </span></span>
<span><span class="co">#&gt; [163] plyr_1.8.9                  parallel_4.3.1             </span></span>
<span><span class="co">#&gt; [165] PRECAST_1.6.5               listenv_0.9.1              </span></span>
<span><span class="co">#&gt; [167] ggrepel_0.9.5               lmom_3.0                   </span></span>
<span><span class="co">#&gt; [169] deldir_2.0-4                Biostrings_2.70.3          </span></span>
<span><span class="co">#&gt; [171] splines_4.3.1               tensor_1.5                 </span></span>
<span><span class="co">#&gt; [173] hms_1.1.3                   igraph_2.0.3               </span></span>
<span><span class="co">#&gt; [175] ggpubr_0.6.0                spatstat.geom_3.3-2        </span></span>
<span><span class="co">#&gt; [177] ggsignif_0.6.4              reshape2_1.4.4             </span></span>
<span><span class="co">#&gt; [179] biomaRt_2.58.2              stats4_4.3.1               </span></span>
<span><span class="co">#&gt; [181] ScaledMatrix_1.10.0         XML_3.99-0.17              </span></span>
<span><span class="co">#&gt; [183] evaluate_0.24.0             httpuv_1.6.15              </span></span>
<span><span class="co">#&gt; [185] RANN_2.6.2                  tidyr_1.3.1                </span></span>
<span><span class="co">#&gt; [187] polyclip_1.10-7             future_1.34.0              </span></span>
<span><span class="co">#&gt; [189] scattermore_1.2             rsvd_1.0.5                 </span></span>
<span><span class="co">#&gt; [191] broom_1.0.6                 xtable_1.8-4               </span></span>
<span><span class="co">#&gt; [193] e1071_1.7-14                rstatix_0.7.2              </span></span>
<span><span class="co">#&gt; [195] later_1.3.2                 viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [197] class_7.3-22                ragg_1.3.2                 </span></span>
<span><span class="co">#&gt; [199] tibble_3.2.1                memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [201] beeswarm_0.4.0              AnnotationDbi_1.64.1       </span></span>
<span><span class="co">#&gt; [203] IRanges_2.36.0              cluster_2.1.4              </span></span>
<span><span class="co">#&gt; [205] globals_0.16.3</span></span></code></pre></div>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Xiao Zhang, Wei Liu, Jin Liu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
